{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "KDM5s = [\"Q3UXZ9\",\"Q38JA7\",\"Q30DN6\",\"Q62240\",\"Q5F3R2\",\"Q80Y84\",\"A1YVX4\",\"Q61T02\",\"Q9VMJ7\",\"P41228\",\"P41229\",\"Q5XUN4\",\"Q23541\",\"P29375\",\"Q9UGL1\",\"P41230\",\"Q9BY66\",\"Q6IQX0\",\"P47156\",\"Q8GUI6\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Try using Bio.Entrez\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "Entrez.email = \"david.brown3@ucsf.edu\"  # Always tell NCBI who you are\n",
    "handle = Entrez.einfo()\n",
    "result = handle.read()\n",
    "handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n",
      "<!DOCTYPE eInfoResult PUBLIC \"-//NLM//DTD einfo 20130322//EN\" \"https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20130322/einfo.dtd\">\n",
      "<eInfoResult>\n",
      "<DbList>\n",
      "\n",
      "\t<DbName>pubmed</DbName>\n",
      "\t<DbName>protein</DbName>\n",
      "\t<DbName>nuccore</DbName>\n",
      "\t<DbName>ipg</DbName>\n",
      "\t<DbName>nucleotide</DbName>\n",
      "\t<DbName>nucgss</DbName>\n",
      "\t<DbName>nucest</DbName>\n",
      "\t<DbName>structure</DbName>\n",
      "\t<DbName>sparcle</DbName>\n",
      "\t<DbName>genome</DbName>\n",
      "\t<DbName>annotinfo</DbName>\n",
      "\t<DbName>assembly</DbName>\n",
      "\t<DbName>bioproject</DbName>\n",
      "\t<DbName>biosample</DbName>\n",
      "\t<DbName>blastdbinfo</DbName>\n",
      "\t<DbName>books</DbName>\n",
      "\t<DbName>cdd</DbName>\n",
      "\t<DbName>clinvar</DbName>\n",
      "\t<DbName>clone</DbName>\n",
      "\t<DbName>gap</DbName>\n",
      "\t<DbName>gapplus</DbName>\n",
      "\t<DbName>grasp</DbName>\n",
      "\t<DbName>dbvar</DbName>\n",
      "\t<DbName>gene</DbName>\n",
      "\t<DbName>gds</DbName>\n",
      "\t<DbName>geoprofiles</DbName>\n",
      "\t<DbName>homologene</DbName>\n",
      "\t<DbName>medgen</DbName>\n",
      "\t<DbName>mesh</DbName>\n",
      "\t<DbName>ncbisearch</DbName>\n",
      "\t<DbName>nlmcatalog</DbName>\n",
      "\t<DbName>omim</DbName>\n",
      "\t<DbName>orgtrack</DbName>\n",
      "\t<DbName>pmc</DbName>\n",
      "\t<DbName>popset</DbName>\n",
      "\t<DbName>probe</DbName>\n",
      "\t<DbName>proteinclusters</DbName>\n",
      "\t<DbName>pcassay</DbName>\n",
      "\t<DbName>biosystems</DbName>\n",
      "\t<DbName>pccompound</DbName>\n",
      "\t<DbName>pcsubstance</DbName>\n",
      "\t<DbName>pubmedhealth</DbName>\n",
      "\t<DbName>seqannot</DbName>\n",
      "\t<DbName>snp</DbName>\n",
      "\t<DbName>sra</DbName>\n",
      "\t<DbName>taxonomy</DbName>\n",
      "\t<DbName>biocollections</DbName>\n",
      "\t<DbName>unigene</DbName>\n",
      "\t<DbName>gencoll</DbName>\n",
      "\t<DbName>gtr</DbName>\n",
      "</DbList>\n",
      "\n",
      "</eInfoResult>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "handle = Entrez.einfo()\n",
    "record = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[u'DbList']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['pubmed', 'protein', 'nuccore', 'ipg', 'nucleotide', 'nucgss', 'nucest', 'structure', 'sparcle', 'genome', 'annotinfo', 'assembly', 'bioproject', 'biosample', 'blastdbinfo', 'books', 'cdd', 'clinvar', 'clone', 'gap', 'gapplus', 'grasp', 'dbvar', 'gene', 'gds', 'geoprofiles', 'homologene', 'medgen', 'mesh', 'ncbisearch', 'nlmcatalog', 'omim', 'orgtrack', 'pmc', 'popset', 'probe', 'proteinclusters', 'pcassay', 'biosystems', 'pccompound', 'pcsubstance', 'pubmedhealth', 'seqannot', 'snp', 'sra', 'taxonomy', 'biocollections', 'unigene', 'gencoll', 'gtr']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbList\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "handle = Entrez.esearch(db=\"protein\", term=\"Q3UXZ9\")\n",
    "record = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['150383496']\n"
     ]
    }
   ],
   "source": [
    "#This give the NCBI id for the correct protein\n",
    "print(record[\"IdList\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'dict' object is not callable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-76f71bd883b6>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mrecord\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mattributes\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: 'dict' object is not callable"
     ]
    }
   ],
   "source": [
    "record.attributes()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "print(record.attributes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Protein sequence record'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#For each of these databases, we can use EInfo again to obtain more information:\n",
    "handle = Entrez.einfo(db=\"protein\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"DbInfo\"][\"Description\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'534686518'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbInfo\"][\"Count\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Try record[\"DbInfo\"].keys() for other information stored in this record. One of the most useful is a list of possible search fields for use with ESearch:\n",
    "\n",
    ">>> for field in record[\"DbInfo\"][\"FieldList\"]:\n",
    "...     print(\"%(Name)s, %(FullName)s, %(Description)s\" % field)\n",
    "...\n",
    "ALL, All Fields, All terms from all searchable fields\n",
    "UID, UID, Unique number assigned to publication\n",
    "FILT, Filter, Limits the records\n",
    "TITL, Title, Words in title of publication\n",
    "WORD, Text Word, Free text associated with publication\n",
    "MESH, MeSH Terms, Medical Subject Headings assigned to publication\n",
    "MAJR, MeSH Major Topic, MeSH terms of major importance to publication\n",
    "AUTH, Author, Author(s) of publication\n",
    "JOUR, Journal, Journal abbreviation of publication\n",
    "AFFL, Affiliation, Author's institutional affiliation and address\n",
    "...\n",
    "That’s a long list, but indirectly this tells you that for the PubMed database, you can do things like Jones[AUTH] to search the author field, or Sanger[AFFL] to restrict to authors at the Sanger Centre. This can be very handy - especially if you are not so familiar with a particular database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[u'Count',\n",
       " u'LastUpdate',\n",
       " u'MenuName',\n",
       " u'Description',\n",
       " u'LinkList',\n",
       " u'DbBuild',\n",
       " u'FieldList',\n",
       " u'DbName']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#What information is stored in the record?\n",
    "record[\"DbInfo\"].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALL, All Fields, All terms from all searchable fields\n",
      "UID, UID, Unique number assigned to each sequence\n",
      "FILT, Filter, Limits the records\n",
      "WORD, Text Word, Free text associated with record\n",
      "TITL, Title, Words in definition line\n",
      "KYWD, Keyword, Nonstandardized terms provided by submitter\n",
      "AUTH, Author, Author(s) of publication\n",
      "JOUR, Journal, Journal abbreviation of publication\n",
      "VOL, Volume, Volume number of publication\n",
      "ISS, Issue, Issue number of publication\n",
      "PAGE, Page Number, Page number(s) of publication\n",
      "ORGN, Organism, Scientific and common names of organism, and all higher levels of taxonomy\n",
      "ACCN, Accession, Accession number of sequence\n",
      "PACC, Primary Accession, Does not include retired secondary accessions\n",
      "GENE, Gene Name, Name of gene associated with sequence\n",
      "PROT, Protein Name, Name of protein associated with sequence\n",
      "ECNO, EC/RN Number, EC number for enzyme or CAS registry number\n",
      "PDAT, Publication Date, Date sequence added to GenBank\n",
      "MDAT, Modification Date, Date of last update\n",
      "SUBS, Substance Name, CAS chemical name or MEDLINE Substance Name\n",
      "PROP, Properties, Classification by source qualifiers and molecule type\n",
      "SQID, SeqID String, String identifier for sequence\n",
      "GPRJ, BioProject, BioProject\n",
      "SLEN, Sequence Length, Length of sequence\n",
      "MLWT, Molecular Weight, Molecular Weight\n",
      "FKEY, Feature key, Feature annotated on sequence\n",
      "PORG, Primary Organism, Scientific and common names of primary organism, and all higher levels of taxonomy\n",
      "ASSM, Assembly, Assembly\n",
      "DIV, Division, Division\n",
      "STRN, Strain, Strain\n",
      "ISOL, Isolate, Isolate\n",
      "CULT, Cultivar, Cultivar\n",
      "BRD, Breed, Breed\n"
     ]
    }
   ],
   "source": [
    "for field in record[\"DbInfo\"][\"FieldList\"]:\n",
    "    print(\"%(Name)s, %(FullName)s, %(Description)s\" % field)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "EFetch: Downloading full records from Entrez\n",
    "\n",
    "EFetch is what you use when you want to retrieve a full record from Entrez. This covers several possible databases, as described on the main EFetch Help page.\n",
    "\n",
    "For most of their databases, the NCBI support several different file formats. Requesting a specific file format from Entrez using Bio.Entrez.efetch() requires specifying the rettype and/or retmode optional arguments. The different combinations are described for each database type on the pages linked to on NCBI efetch webpage (e.g. literature, sequences and taxonomy).\n",
    "\n",
    "One common usage is downloading sequences in the FASTA or GenBank/GenPept plain text formats (which can then be parsed with Bio.SeqIO, see Sections 5.3.1 and 9.6). From the Cypripedioideae example above, we can download GenBank record EU490707 using Bio.Entrez.efetch:\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       EU490707                1302 bp    DNA     linear   PLN 26-JUL-2016\n",
      "DEFINITION  Selenipedium aequinoctiale maturase K (matK) gene, partial cds;\n",
      "            chloroplast.\n",
      "ACCESSION   EU490707\n",
      "VERSION     EU490707.1\n",
      "KEYWORDS    .\n",
      "SOURCE      chloroplast Selenipedium aequinoctiale\n",
      "  ORGANISM  Selenipedium aequinoctiale\n",
      "            Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;\n",
      "            Spermatophyta; Magnoliophyta; Liliopsida; Asparagales; Orchidaceae;\n",
      "            Cypripedioideae; Selenipedium.\n",
      "REFERENCE   1  (bases 1 to 1302)\n",
      "  AUTHORS   Neubig,K.M., Whitten,W.M., Carlsward,B.S., Blanco,M.A., Endara,L.,\n",
      "            Williams,N.H. and Moore,M.\n",
      "  TITLE     Phylogenetic utility of ycf1 in orchids: a plastid gene more\n",
      "            variable than matK\n",
      "  JOURNAL   Plant Syst. Evol. 277 (1-2), 75-84 (2009)\n",
      "REFERENCE   2  (bases 1 to 1302)\n",
      "  AUTHORS   Neubig,K.M., Whitten,W.M., Carlsward,B.S., Blanco,M.A.,\n",
      "            Endara,C.L., Williams,N.H. and Moore,M.J.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (14-FEB-2008) Department of Botany, University of\n",
      "            Florida, 220 Bartram Hall, Gainesville, FL 32611-8526, USA\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..1302\n",
      "                     /organism=\"Selenipedium aequinoctiale\"\n",
      "                     /organelle=\"plastid:chloroplast\"\n",
      "                     /mol_type=\"genomic DNA\"\n",
      "                     /specimen_voucher=\"FLAS:Blanco 2475\"\n",
      "                     /db_xref=\"taxon:256374\"\n",
      "     gene            <1..>1302\n",
      "                     /gene=\"matK\"\n",
      "     CDS             <1..>1302\n",
      "                     /gene=\"matK\"\n",
      "                     /codon_start=1\n",
      "                     /transl_table=11\n",
      "                     /product=\"maturase K\"\n",
      "                     /protein_id=\"ACC99456.1\"\n",
      "                     /translation=\"IFYEPVEIFGYDNKSSLVLVKRLITRMYQQNFLISSVNDSNQKG\n",
      "                     FWGHKHFFSSHFSSQMVSEGFGVILEIPFSSQLVSSLEEKKIPKYQNLRSIHSIFPFL\n",
      "                     EDKFLHLNYVSDLLIPHPIHLEILVQILQCRIKDVPSLHLLRLLFHEYHNLNSLITSK\n",
      "                     KFIYAFSKRKKRFLWLLYNSYVYECEYLFQFLRKQSSYLRSTSSGVFLERTHLYVKIE\n",
      "                     HLLVVCCNSFQRILCFLKDPFMHYVRYQGKAILASKGTLILMKKWKFHLVNFWQSYFH\n",
      "                     FWSQPYRIHIKQLSNYSFSFLGYFSSVLENHLVVRNQMLENSFIINLLTKKFDTIAPV\n",
      "                     ISLIGSLSKAQFCTVLGHPISKPIWTDFSDSDILDRFCRICRNLCRYHSGSSKKQVLY\n",
      "                     RIKYILRLSCARTLARKHKSTVRTFMRRLGSGLLEEFFMEEE\"\n",
      "ORIGIN      \n",
      "        1 attttttacg aacctgtgga aatttttggt tatgacaata aatctagttt agtacttgtg\n",
      "       61 aaacgtttaa ttactcgaat gtatcaacag aattttttga tttcttcggt taatgattct\n",
      "      121 aaccaaaaag gattttgggg gcacaagcat tttttttctt ctcatttttc ttctcaaatg\n",
      "      181 gtatcagaag gttttggagt cattctggaa attccattct cgtcgcaatt agtatcttct\n",
      "      241 cttgaagaaa aaaaaatacc aaaatatcag aatttacgat ctattcattc aatatttccc\n",
      "      301 tttttagaag acaaattttt acatttgaat tatgtgtcag atctactaat accccatccc\n",
      "      361 atccatctgg aaatcttggt tcaaatcctt caatgccgga tcaaggatgt tccttctttg\n",
      "      421 catttattgc gattgctttt ccacgaatat cataatttga atagtctcat tacttcaaag\n",
      "      481 aaattcattt acgccttttc aaaaagaaag aaaagattcc tttggttact atataattct\n",
      "      541 tatgtatatg aatgcgaata tctattccag tttcttcgta aacagtcttc ttatttacga\n",
      "      601 tcaacatctt ctggagtctt tcttgagcga acacatttat atgtaaaaat agaacatctt\n",
      "      661 ctagtagtgt gttgtaattc ttttcagagg atcctatgct ttctcaagga tcctttcatg\n",
      "      721 cattatgttc gatatcaagg aaaagcaatt ctggcttcaa agggaactct tattctgatg\n",
      "      781 aagaaatgga aatttcatct tgtgaatttt tggcaatctt attttcactt ttggtctcaa\n",
      "      841 ccgtatagga ttcatataaa gcaattatcc aactattcct tctcttttct ggggtatttt\n",
      "      901 tcaagtgtac tagaaaatca tttggtagta agaaatcaaa tgctagagaa ttcatttata\n",
      "      961 ataaatcttc tgactaagaa attcgatacc atagccccag ttatttctct tattggatca\n",
      "     1021 ttgtcgaaag ctcaattttg tactgtattg ggtcatccta ttagtaaacc gatctggacc\n",
      "     1081 gatttctcgg attctgatat tcttgatcga ttttgccgga tatgtagaaa tctttgtcgt\n",
      "     1141 tatcacagcg gatcctcaaa aaaacaggtt ttgtatcgta taaaatatat acttcgactt\n",
      "     1201 tcgtgtgcta gaactttggc acggaaacat aaaagtacag tacgcacttt tatgcgaaga\n",
      "     1261 ttaggttcgg gattattaga agaattcttt atggaagaag aa\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Cookbook example\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id=\"EU490707\", rettype=\"gb\", retmode=\"text\")\n",
    "print(handle.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       KDM5A_MOUSE             1690 aa            linear   ROD 28-MAR-2018\n",
      "DEFINITION  RecName: Full=Lysine-specific demethylase 5A; AltName: Full=Histone\n",
      "            demethylase JARID1A; AltName: Full=Jumonji/ARID domain-containing\n",
      "            protein 1A; AltName: Full=Retinoblastoma-binding protein 2;\n",
      "            Short=RBBP-2.\n",
      "ACCESSION   Q3UXZ9\n",
      "VERSION     Q3UXZ9.2\n",
      "DBSOURCE    UniProtKB: locus KDM5A_MOUSE, accession Q3UXZ9;\n",
      "            class: standard.\n",
      "            extra accessions:Q3TM94,Q3UMI5,Q66JZ3\n",
      "            created: Jun 26, 2007.\n",
      "            sequence updated: Jun 26, 2007.\n",
      "            annotation updated: Mar 28, 2018.\n",
      "            xrefs: AC155720.7, AC078896.23, AK135085.1, BAE22414.1, AK144877.1,\n",
      "            BAE26113.1, AK166055.1, BAE38548.1, BC080691.1, AAH80691.1,\n",
      "            NP_666109.2\n",
      "            xrefs (non-sequence databases): CCDS:CCDS51889.1,\n",
      "            UniGene:Mm.404761, UniGene:Mm.463658, ProteinModelPortal:Q3UXZ9,\n",
      "            SMR:Q3UXZ9, BioGrid:229572, IntAct:Q3UXZ9,\n",
      "            STRING:10090.ENSMUSP00000098558, BindingDB:Q3UXZ9,\n",
      "            ChEMBL:CHEMBL3707461, iPTMnet:Q3UXZ9, PhosphoSitePlus:Q3UXZ9,\n",
      "            EPD:Q3UXZ9, MaxQB:Q3UXZ9, PaxDb:Q3UXZ9, PeptideAtlas:Q3UXZ9,\n",
      "            PRIDE:Q3UXZ9, Ensembl:ENSMUST00000005108,\n",
      "            Ensembl:ENSMUSP00000005108, Ensembl:ENSMUSG00000030180,\n",
      "            GeneID:214899, KEGG:mmu:214899, UCSC:uc009dne.2, CTD:5927,\n",
      "            MGI:2136980, eggNOG:KOG1246, eggNOG:ENOG410XR9J,\n",
      "            GeneTree:ENSGT00530000063118, HOGENOM:HOG000290719,\n",
      "            InParanoid:Q3UXZ9, KO:K11446, OMA:CHEWYHG, PhylomeDB:Q3UXZ9,\n",
      "            TreeFam:TF106476, BRENDA:1.14.11.B2, Reactome:R-MMU-3214842,\n",
      "            ChiTaRS:Kdm5a, PRO:PR:Q3UXZ9, Proteomes:UP000000589,\n",
      "            Bgee:ENSMUSG00000030180, CleanEx:MM_RBP2, ExpressionAtlas:Q3UXZ9,\n",
      "            Genevisible:Q3UXZ9, GO:0019907, GO:0005730, GO:0005634, GO:0032993,\n",
      "            GO:0003682, GO:0031490, GO:0001046, GO:0051213, GO:0003677,\n",
      "            GO:0003700, GO:0042393, GO:0034648, GO:0034647, GO:0035064,\n",
      "            GO:0003713, GO:0008270, GO:0032922, GO:0034720, GO:0008584,\n",
      "            GO:1901726, GO:0000122, GO:0045893, GO:0051090, GO:0007283,\n",
      "            GO:0006351, Gene3D:1.10.150.60, Gene3D:3.30.40.10,\n",
      "            InterPro:IPR001606, InterPro:IPR036431, InterPro:IPR003347,\n",
      "            InterPro:IPR003349, InterPro:IPR013637, InterPro:IPR019786,\n",
      "            InterPro:IPR004198, InterPro:IPR011011, InterPro:IPR001965,\n",
      "            InterPro:IPR019787, InterPro:IPR013083, Pfam:PF01388, Pfam:PF02373,\n",
      "            Pfam:PF02375, Pfam:PF00628, Pfam:PF08429, Pfam:PF02928,\n",
      "            SMART:SM00501, SMART:SM00558, SMART:SM00545, SMART:SM00249,\n",
      "            SUPFAM:SSF46774, SUPFAM:SSF57903, PROSITE:PS51011, PROSITE:PS51184,\n",
      "            PROSITE:PS51183, PROSITE:PS01359, PROSITE:PS50016\n",
      "KEYWORDS    Activator; Biological rhythms; Chromatin regulator; Complete\n",
      "            proteome; Developmental protein; Dioxygenase; Direct protein\n",
      "            sequencing; Iron; Isopeptide bond; Metal-binding; Nucleus;\n",
      "            Oxidoreductase; Phosphoprotein; Reference proteome; Repeat;\n",
      "            Transcription; Transcription regulation; Ubl conjugation; Zinc;\n",
      "            Zinc-finger.\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1  (residues 1 to 1690)\n",
      "  AUTHORS   Church,D.M., Goodstadt,L., Hillier,L.W., Zody,M.C., Goldstein,S.,\n",
      "            She,X., Bult,C.J., Agarwala,R., Cherry,J.L., DiCuccio,M.,\n",
      "            Hlavina,W., Kapustin,Y., Meric,P., Maglott,D., Birtle,Z.,\n",
      "            Marques,A.C., Graves,T., Zhou,S., Teague,B., Potamousis,K.,\n",
      "            Churas,C., Place,M., Herschleb,J., Runnheim,R., Forrest,D.,\n",
      "            Amos-Landgraf,J., Schwartz,D.C., Cheng,Z., Lindblad-Toh,K.,\n",
      "            Eichler,E.E. and Ponting,C.P.\n",
      "  CONSRTM   Mouse Genome Sequencing Consortium\n",
      "  TITLE     Lineage-specific biology revealed by a finished genome assembly of\n",
      "            the mouse\n",
      "  JOURNAL   PLoS Biol. 7 (5), e1000112 (2009)\n",
      "   PUBMED   19468303\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA].;\n",
      "            STRAIN=C57BL/6J\n",
      "REFERENCE   2  (residues 1 to 1690)\n",
      "  AUTHORS   Carninci,P., Kasukawa,T., Katayama,S., Gough,J., Frith,M.C.,\n",
      "            Maeda,N., Oyama,R., Ravasi,T., Lenhard,B., Wells,C., Kodzius,R.,\n",
      "            Shimokawa,K., Bajic,V.B., Brenner,S.E., Batalov,S., Forrest,A.R.,\n",
      "            Zavolan,M., Davis,M.J., Wilming,L.G., Aidinis,V., Allen,J.E.,\n",
      "            Ambesi-Impiombato,A., Apweiler,R., Aturaliya,R.N., Bailey,T.L.,\n",
      "            Bansal,M., Baxter,L., Beisel,K.W., Bersano,T., Bono,H., Chalk,A.M.,\n",
      "            Chiu,K.P., Choudhary,V., Christoffels,A., Clutterbuck,D.R.,\n",
      "            Crowe,M.L., Dalla,E., Dalrymple,B.P., de Bono,B., Della Gatta,G.,\n",
      "            di Bernardo,D., Down,T., Engstrom,P., Fagiolini,M., Faulkner,G.,\n",
      "            Fletcher,C.F., Fukushima,T., Furuno,M., Futaki,S., Gariboldi,M.,\n",
      "            Georgii-Hemming,P., Gingeras,T.R., Gojobori,T., Green,R.E.,\n",
      "            Gustincich,S., Harbers,M., Hayashi,Y., Hensch,T.K., Hirokawa,N.,\n",
      "            Hill,D., Huminiecki,L., Iacono,M., Ikeo,K., Iwama,A., Ishikawa,T.,\n",
      "            Jakt,M., Kanapin,A., Katoh,M., Kawasawa,Y., Kelso,J., Kitamura,H.,\n",
      "            Kitano,H., Kollias,G., Krishnan,S.P., Kruger,A., Kummerfeld,S.K.,\n",
      "            Kurochkin,I.V., Lareau,L.F., Lazarevic,D., Lipovich,L., Liu,J.,\n",
      "            Liuni,S., McWilliam,S., Madan Babu,M., Madera,M., Marchionni,L.,\n",
      "            Matsuda,H., Matsuzawa,S., Miki,H., Mignone,F., Miyake,S.,\n",
      "            Morris,K., Mottagui-Tabar,S., Mulder,N., Nakano,N., Nakauchi,H.,\n",
      "            Ng,P., Nilsson,R., Nishiguchi,S., Nishikawa,S., Nori,F., Ohara,O.,\n",
      "            Okazaki,Y., Orlando,V., Pang,K.C., Pavan,W.J., Pavesi,G.,\n",
      "            Pesole,G., Petrovsky,N., Piazza,S., Reed,J., Reid,J.F., Ring,B.Z.,\n",
      "            Ringwald,M., Rost,B., Ruan,Y., Salzberg,S.L., Sandelin,A.,\n",
      "            Schneider,C., Schonbach,C., Sekiguchi,K., Semple,C.A., Seno,S.,\n",
      "            Sessa,L., Sheng,Y., Shibata,Y., Shimada,H., Shimada,K., Silva,D.,\n",
      "            Sinclair,B., Sperling,S., Stupka,E., Sugiura,K., Sultana,R.,\n",
      "            Takenaka,Y., Taki,K., Tammoja,K., Tan,S.L., Tang,S., Taylor,M.S.,\n",
      "            Tegner,J., Teichmann,S.A., Ueda,H.R., van Nimwegen,E., Verardo,R.,\n",
      "            Wei,C.L., Yagi,K., Yamanishi,H., Zabarovsky,E., Zhu,S., Zimmer,A.,\n",
      "            Hide,W., Bult,C., Grimmond,S.M., Teasdale,R.D., Liu,E.T.,\n",
      "            Brusic,V., Quackenbush,J., Wahlestedt,C., Mattick,J.S., Hume,D.A.,\n",
      "            Kai,C., Sasaki,D., Tomaru,Y., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Suzuki,M., Aoki,J., Arakawa,T., Iida,J., Imamura,K., Itoh,M.,\n",
      "            Kato,T., Kawaji,H., Kawagashira,N., Kawashima,T., Kojima,M.,\n",
      "            Kondo,S., Konno,H., Nakano,K., Ninomiya,N., Nishio,T., Okada,M.,\n",
      "            Plessy,C., Shibata,K., Shiraki,T., Suzuki,S., Tagami,M., Waki,K.,\n",
      "            Watahiki,A., Okamura-Oho,Y., Suzuki,H., Kawai,J. and Hayashizaki,Y.\n",
      "  CONSRTM   FANTOM Consortium; RIKEN Genome Exploration Research Group and\n",
      "            Genome Science Group (Genome Network Project Core Group)\n",
      "  TITLE     The transcriptional landscape of the mammalian genome\n",
      "  JOURNAL   Science 309 (5740), 1559-1563 (2005)\n",
      "   PUBMED   16141072\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] OF 1-1552.;\n",
      "            STRAIN=C57BL/6J; TISSUE=Lung, and Olfactory bulb\n",
      "            Erratum:[Science. 2006 Mar 24;311(5768):1713]\n",
      "REFERENCE   3  (residues 1 to 1690)\n",
      "  AUTHORS   Gerhard,D.S., Wagner,L., Feingold,E.A., Shenmen,C.M., Grouse,L.H.,\n",
      "            Schuler,G., Klein,S.L., Old,S., Rasooly,R., Good,P., Guyer,M.,\n",
      "            Peck,A.M., Derge,J.G., Lipman,D., Collins,F.S., Jang,W., Sherry,S.,\n",
      "            Feolo,M., Misquitta,L., Lee,E., Rotmistrovsky,K., Greenhut,S.F.,\n",
      "            Schaefer,C.F., Buetow,K., Bonner,T.I., Haussler,D., Kent,J.,\n",
      "            Kiekhaus,M., Furey,T., Brent,M., Prange,C., Schreiber,K.,\n",
      "            Shapiro,N., Bhat,N.K., Hopkins,R.F., Hsie,F., Driscoll,T.,\n",
      "            Soares,M.B., Casavant,T.L., Scheetz,T.E., Brown-stein,M.J.,\n",
      "            Usdin,T.B., Toshiyuki,S., Carninci,P., Piao,Y., Dudekula,D.B.,\n",
      "            Ko,M.S., Kawakami,K., Suzuki,Y., Sugano,S., Gruber,C.E.,\n",
      "            Smith,M.R., Simmons,B., Moore,T., Waterman,R., Johnson,S.L.,\n",
      "            Ruan,Y., Wei,C.L., Mathavan,S., Gunaratne,P.H., Wu,J., Garcia,A.M.,\n",
      "            Hulyk,S.W., Fuh,E., Yuan,Y., Sneed,A., Kowis,C., Hodgson,A.,\n",
      "            Muzny,D.M., McPherson,J., Gibbs,R.A., Fahey,J., Helton,E.,\n",
      "            Ketteman,M., Madan,A., Rodrigues,S., Sanchez,A., Whiting,M.,\n",
      "            Madari,A., Young,A.C., Wetherby,K.D., Granite,S.J., Kwong,P.N.,\n",
      "            Brinkley,C.P., Pearson,R.L., Bouffard,G.G., Blakesly,R.W.,\n",
      "            Green,E.D., Dickson,M.C., Rodriguez,A.C., Grimwood,J., Schmutz,J.,\n",
      "            Myers,R.M., Butterfield,Y.S., Griffith,M., Griffith,O.L.,\n",
      "            Krzywinski,M.I., Liao,N., Morin,R., Palmquist,D., Petrescu,A.S.,\n",
      "            Skalska,U., Smailus,D.E., Stott,J.M., Schnerch,A., Schein,J.E.,\n",
      "            Jones,S.J., Holt,R.A., Baross,A., Marra,M.A., Clifton,S.,\n",
      "            Makowski,K.A., Bosak,S. and Malek,J.\n",
      "  CONSRTM   MGC Project Team\n",
      "  TITLE     The status, quality, and expansion of the NIH full-length cDNA\n",
      "            project: the Mammalian Gene Collection (MGC)\n",
      "  JOURNAL   Genome Res. 14 (10B), 2121-2127 (2004)\n",
      "   PUBMED   15489334\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] OF 1-1099.;\n",
      "            STRAIN=C57BL/6J; TISSUE=Germ cell\n",
      "            Erratum:[Genome Res. 2006 Jun;16(6):804. Morrin, Ryan [corrected to\n",
      "            Morin, Ryan]]\n",
      "REFERENCE   4  (residues 1 to 1690)\n",
      "  AUTHORS   Lubec,G., Sunyer,B. and Chen,W.-Q.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-JAN-2009) to UniProtKB\n",
      "  REMARK    PROTEIN SEQUENCE OF 1538-1546, AND IDENTIFICATION BY MASS\n",
      "            SPECTROMETRY.;\n",
      "            STRAIN=OF1; TISSUE=Hippocampus\n",
      "REFERENCE   5  (residues 1 to 1690)\n",
      "  AUTHORS   Klose,R.J., Yan,Q., Tothova,Z., Yamane,K., Erdjument-Bromage,H.,\n",
      "            Tempst,P., Gilliland,D.G., Zhang,Y. and Kaelin,W.G. Jr.\n",
      "  TITLE     The retinoblastoma binding protein RBP2 is an H3K4 demethylase\n",
      "  JOURNAL   Cell 128 (5), 889-900 (2007)\n",
      "   PUBMED   17320163\n",
      "  REMARK    FUNCTION, AND DISRUPTION PHENOTYPE.\n",
      "REFERENCE   6  (residues 1 to 1690)\n",
      "  AUTHORS   Christensen,J., Agger,K., Cloos,P.A., Pasini,D., Rose,S.,\n",
      "            Sennels,L., Rappsilber,J., Hansen,K.H., Salcini,A.E. and Helin,K.\n",
      "  TITLE     RBP2 belongs to a family of demethylases, specific for tri-and\n",
      "            dimethylated lysine 4 on histone 3\n",
      "  JOURNAL   Cell 128 (6), 1063-1076 (2007)\n",
      "   PUBMED   17320161\n",
      "  REMARK    FUNCTION.\n",
      "REFERENCE   7  (residues 1 to 1690)\n",
      "  AUTHORS   Peng,J.C., Valouev,A., Swigut,T., Zhang,J., Zhao,Y., Sidow,A. and\n",
      "            Wysocka,J.\n",
      "  TITLE     Jarid2/Jumonji coordinates control of PRC2 enzymatic activity and\n",
      "            target gene occupancy in pluripotent cells\n",
      "  JOURNAL   Cell 139 (7), 1290-1302 (2009)\n",
      "   PUBMED   20064375\n",
      "  REMARK    SUBCELLULAR LOCATION, DOMAIN GSGFP MOTIF, AND INTERACTION WITH\n",
      "            SUZ12.\n",
      "REFERENCE   8  (residues 1 to 1690)\n",
      "  AUTHORS   Trost,M., English,L., Lemieux,S., Courcelles,M., Desjardins,M. and\n",
      "            Thibault,P.\n",
      "  TITLE     The phagosomal proteome in interferon-gamma-activated macrophages\n",
      "  JOURNAL   Immunity 30 (1), 143-154 (2009)\n",
      "   PUBMED   19144319\n",
      "  REMARK    PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-1111, AND\n",
      "            IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].\n",
      "REFERENCE   9  (residues 1 to 1690)\n",
      "  AUTHORS   Huttlin,E.L., Jedrychowski,M.P., Elias,J.E., Goswami,T., Rad,R.,\n",
      "            Beausoleil,S.A., Villen,J., Haas,W., Sowa,M.E. and Gygi,S.P.\n",
      "  TITLE     A tissue-specific atlas of mouse protein phosphorylation and\n",
      "            expression\n",
      "  JOURNAL   Cell 143 (7), 1174-1189 (2010)\n",
      "   PUBMED   21183079\n",
      "  REMARK    PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-1111; THR-1343;\n",
      "            SER-1345; TYR-1595; SER-1598; SER-1603 AND SER-1666, AND\n",
      "            IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].;\n",
      "            TISSUE=Brain, Kidney, Lung, Spleen, and Testis\n",
      "REFERENCE   10 (residues 1 to 1690)\n",
      "  AUTHORS   DiTacchio,L., Le,H.D., Vollmers,C., Hatori,M., Witcher,M.,\n",
      "            Secombe,J. and Panda,S.\n",
      "  TITLE     Histone lysine demethylase JARID1a activates CLOCK-BMAL1 and\n",
      "            influences the circadian clock\n",
      "  JOURNAL   Science 333 (6051), 1881-1885 (2011)\n",
      "   PUBMED   21960634\n",
      "  REMARK    FUNCTION, AND INTERACTION WITH ARNTL; CLOCK AND HDAC1.\n",
      "COMMENT     On or before Jul 3, 2007 this sequence version replaced\n",
      "            gi:123783469, gi:123784837, gi:81884291, gi:123795350.\n",
      "            [FUNCTION] Histone demethylase that specifically demethylates\n",
      "            'Lys-4' of histone H3, thereby playing a central role in histone\n",
      "            code. Does not demethylate histone H3 'Lys-9', H3 'Lys-27', H3\n",
      "            'Lys-36', H3 'Lys-79' or H4 'Lys-20'. Demethylates trimethylated\n",
      "            and dimethylated but not monomethylated H3 'Lys-4'\n",
      "            (PubMed:17320161, PubMed:17320163). Regulates specific gene\n",
      "            transcription through DNA-binding on 5'-CCGCCC-3' motif. May\n",
      "            stimulate transcription mediated by nuclear receptors. Involved in\n",
      "            transcriptional regulation of Hox proteins during cell\n",
      "            differentiation (By similarity). May participate in transcriptional\n",
      "            repression of cytokines such as CXCL12. Plays a role in the\n",
      "            regulation of the circadian rhythm and in maintaining the normal\n",
      "            periodicity of the circadian clock. In a histone\n",
      "            demethylase-independent manner, acts as a coactivator of the\n",
      "            CLOCK-ARNTL/BMAL1-mediated transcriptional activation of PER1/2 and\n",
      "            other clock-controlled genes and increases histone acetylation at\n",
      "            PER1/2 promoters by inhibiting the activity of HDAC1\n",
      "            (PubMed:21960634). Seems to act as a transcriptional corepressor\n",
      "            for some genes such as MT1F and to favor the proliferation of\n",
      "            cancer cells (By similarity). {ECO:0000250|UniProtKB:P29375,\n",
      "            ECO:0000269|PubMed:17320161, ECO:0000269|PubMed:17320163,\n",
      "            ECO:0000269|PubMed:21960634}.\n",
      "            [COFACTOR] Name=Fe(2+); Xref=ChEBI:CHEBI:29033;\n",
      "            Evidence={ECO:0000250}; Note=Binds 1 Fe(2+) ion per subunit.\n",
      "            {ECO:0000250}.\n",
      "            [SUBUNIT] Interacts with RB1, ESR1, MYC, MYCN and LMO2 (By\n",
      "            similarity). Interacts with SUZ12; the interaction is direct\n",
      "            (PubMed:20064375). Interacts with HDAC1, ARNTL/BMAL1 and CLOCK\n",
      "            (PubMed:21960634). Interacts (via PHD-type 1 zinc finger) with\n",
      "            histone H3 unmodified at 'Lys-4' and (via PHD-type 3 zinc finger)\n",
      "            with histone H3 di- and trimethylated at 'Lys-4' (By similarity).\n",
      "            {ECO:0000250|UniProtKB:P29375, ECO:0000269|PubMed:20064375,\n",
      "            ECO:0000269|PubMed:21960634}.\n",
      "            [INTERACTION] Q80U70:Suz12; NbExp=2; IntAct=EBI-2531441,\n",
      "            EBI-2526494.\n",
      "            [SUBCELLULAR LOCATION] Nucleus, nucleolus\n",
      "            {ECO:0000250|UniProtKB:P29375}. Nucleus\n",
      "            {ECO:0000255|PROSITE-ProRule:PRU00355,\n",
      "            ECO:0000255|PROSITE-ProRule:PRU00537, ECO:0000269|PubMed:20064375}.\n",
      "            Note=Occupies promoters of genes involved in RNA metabolism and\n",
      "            mitochondrial function. {ECO:0000269|PubMed:20064375}.\n",
      "            [DOMAIN] The GSGFP motif is required for the interaction with SUZ12\n",
      "            (PubMed:20064375). The ARID domain specifically binds to the CCGCCC\n",
      "            motif and is required for the lysine-specific histone demethylase\n",
      "            activity. The PHD-type 3 zinc finger is required for the\n",
      "            interaction with histone H3 di- and trimethylated at 'Lys-4' (By\n",
      "            similarity). {ECO:0000250|UniProtKB:P29375,\n",
      "            ECO:0000269|PubMed:20064375}.\n",
      "            [DISRUPTION PHENOTYPE] Mice are grossly normal, except that they\n",
      "            exhibit behavioral abnormalities when held upside down by the tail,\n",
      "            and slight hematological defects. {ECO:0000269|PubMed:17320163}.\n",
      "            [SIMILARITY] Belongs to the JARID1 histone demethylase family.\n",
      "            {ECO:0000305}.\n",
      "            [SEQUENCE CAUTION] Sequence=BAE22414.1; Type=Erroneous initiation;\n",
      "            Note=Translation N-terminally extended.; Evidence={ECO:0000305}.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..1690\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "     gene            1..1690\n",
      "                     /gene=\"Kdm5a\"\n",
      "                     /gene_synonym=\"Jarid1a\"\n",
      "                     /gene_synonym=\"Rbp2\"\n",
      "     Protein         1..1690\n",
      "                     /product=\"Lysine-specific demethylase 5A\"\n",
      "                     /EC_number=\"1.14.11.-\"\n",
      "                     /note=\"Histone demethylase JARID1A; Jumonji/ARID\n",
      "                     domain-containing protein 1A; Retinoblastoma-binding\n",
      "                     protein 2; RBBP-2\"\n",
      "                     /UniProtKB_evidence=\"Evidence at protein level\"\n",
      "     Region          1..1690\n",
      "                     /region_name=\"Mature chain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Lysine-specific demethylase 5A.\n",
      "                     /FTId=PRO_0000292411.\"\n",
      "     Region          10\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"A -> S (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          14\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"V -> R (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          17\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"P -> A (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          18..59\n",
      "                     /region_name=\"JmjN\"\n",
      "                     /note=\"Small domain found in the jumonji family of\n",
      "                     transcription factors; smart00545\"\n",
      "                     /db_xref=\"CDD:128818\"\n",
      "     Region          19..60\n",
      "                     /region_name=\"Domain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"JmjN. {ECO:0000255|PROSITE-ProRule:PRU00537}.\"\n",
      "     Region          24\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"P -> A (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          84..174\n",
      "                     /region_name=\"Domain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"ARID. {ECO:0000255|PROSITE-ProRule:PRU00355}.\"\n",
      "     Region          85..170\n",
      "                     /region_name=\"ARID\"\n",
      "                     /note=\"ARID/BRIGHT DNA binding domain; smart01014\"\n",
      "                     /db_xref=\"CDD:198082\"\n",
      "     Bond            bond(191)\n",
      "                     /bond_type=\"xlink\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
      "                     G-Cter in SUMO2). {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            204\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          293..343\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"PHD-type 1.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00146}.\"\n",
      "     Region          295..343\n",
      "                     /region_name=\"PHD1_KDM5A\"\n",
      "                     /note=\"PHD finger 1 found in Lysine-specific demethylase\n",
      "                     5A (KDM5A); cd15602\"\n",
      "                     /db_xref=\"CDD:277075\"\n",
      "     Site            order(295,307..311,315,335)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"putative histone H3 binding site [polypeptide\n",
      "                     binding]\"\n",
      "                     /db_xref=\"CDD:277075\"\n",
      "     Site            order(296,299,311,314,319,322,337,340)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"Zn binding site [ion binding]\"\n",
      "                     /db_xref=\"CDD:277075\"\n",
      "     Site            409\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          419..423\n",
      "                     /region_name=\"Short sequence motif of biological interest\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"GSGFP motif. {ECO:0000269|PubMed:20064375}.\"\n",
      "     Region          437..603\n",
      "                     /region_name=\"Domain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"JmjC. {ECO:0000255|PROSITE-ProRule:PRU00538}.\"\n",
      "     Region          470..586\n",
      "                     /region_name=\"JmjC\"\n",
      "                     /note=\"JmjC domain, hydroxylase; pfam02373\"\n",
      "                     /db_xref=\"CDD:202224\"\n",
      "     Site            483\n",
      "                     /site_type=\"metal-binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Iron; catalytic.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00538}.\"\n",
      "     Site            485\n",
      "                     /site_type=\"metal-binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Iron; catalytic. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            491\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            493\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            501\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          544\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"V -> A (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Site            571\n",
      "                     /site_type=\"metal-binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Iron; catalytic.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00538}.\"\n",
      "     Region          676..728\n",
      "                     /region_name=\"zf-C5HC2\"\n",
      "                     /note=\"C5HC2 zinc finger; pfam02928\"\n",
      "                     /db_xref=\"CDD:280996\"\n",
      "     Region          676..728\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"C5HC2. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          741..1070\n",
      "                     /region_name=\"PLU-1\"\n",
      "                     /note=\"PLU-1-like protein; pfam08429\"\n",
      "                     /db_xref=\"CDD:285609\"\n",
      "     Bond            bond(1007)\n",
      "                     /bond_type=\"xlink\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
      "                     G-Cter in SUMO2). {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1111\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:19144319,\n",
      "                     ECO:0000244|PubMed:21183079}.\"\n",
      "     Region          1153..1210\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"PHD-type 2.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00146}.\"\n",
      "     Region          1163..1215\n",
      "                     /region_name=\"PHD2_KDM5A\"\n",
      "                     /note=\"PHD finger 2 found in Lysine-specific demethylase\n",
      "                     5A (KDM5A); cd15606\"\n",
      "                     /db_xref=\"CDD:277079\"\n",
      "     Site            order(1163,1173..1177,1181,1210)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"putative histone H3 binding site [polypeptide\n",
      "                     binding]\"\n",
      "                     /db_xref=\"CDD:277079\"\n",
      "     Site            order(1164,1166,1177,1180,1185,1188,1212,1215)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"Zn binding site [ion binding]\"\n",
      "                     /db_xref=\"CDD:277079\"\n",
      "     Region          1272\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"A -> P (in Ref. 2; BAE22414). {ECO:0000305}.\"\n",
      "     Site            1330\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1331\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1343\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphothreonine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Site            1345\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Site            1438\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          1484..1579\n",
      "                     /region_name=\"Compositionally biased region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Lys-rich.\"\n",
      "     Site            1488\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1595\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphotyrosine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Site            1598\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Region          1599..1653\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"PHD-type 3.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00146}.\"\n",
      "     Site            1603\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Region          1608..1659\n",
      "                     /region_name=\"PHD3_KDM5A\"\n",
      "                     /note=\"PHD finger 3 found in Lysine-specific demethylase\n",
      "                     5A (KDM5A); cd15686\"\n",
      "                     /db_xref=\"CDD:277156\"\n",
      "     Site            order(1610,1615,1628,1632,1637,1640,1655,1658)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"Zn binding site [ion binding]\"\n",
      "                     /db_xref=\"CDD:277156\"\n",
      "     Region          1622..1690\n",
      "                     /region_name=\"Region of interest in the sequence\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Interaction with LMO2.\n",
      "                     {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            order(1624..1629,1633,1635,1638)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"H3K4me3 binding site [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:277156\"\n",
      "     Site            1666\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "ORIGIN      \n",
      "        1 masvgpggya aefvpppecp vfepsweeft dplsfigrir pfaektgick irppkdwqpp\n",
      "       61 facevktfrf tprvqrlnel eamtrvrldf ldqlakfwel qgstlkipvv erkildlyal\n",
      "      121 skivaskggf eivtkekkws kvgsrlgylp gkgtgsllks hyerilypye lfqsgvslmg\n",
      "      181 vqmpdldlke kveaevlstd iqpspergtr mnippkrtrr vksqsdsgev nrntelkklq\n",
      "      241 ifgagpkvvg lavgakdked evtrrrkvtn rsdafnmqmr qrkgtlsvnf vdlyvcmfcg\n",
      "      301 rgnnedklll cdgcddsyht fcllpplpdv pkgdwrcpkc vaeecnkpre afgfeqavre\n",
      "      361 ytlqsfgema dnfksdyfnm pvhmvptelv ekefwrlvss ieedviveyg adisskdfgs\n",
      "      421 gfpkkdgqrk mlpeeeeyal sgwnlnnmpv leqsvlahin vdisgmkvpw lyvgmcfssf\n",
      "      481 cwhiedhwsy sinylhwgep ktwygvpsha aeqleevmre lapelfesqp dllhqlvtim\n",
      "      541 npnvlmehgv pvyrtnqcag efvvtfpray hsgfnqgynf aeavnfctad wlpigrqcvn\n",
      "      601 hyrrlrrhcv fsheelifkm aadpecldvg laamvckelt lmteeetrlr esvvqmgvvm\n",
      "      661 seeevfelvp dderqcsacr ttcflsaltc scnperlvcl yhptdlcscp mqnkclryry\n",
      "      721 pledlpslly gvkvraqsyd twvnrvteal sasfnhkkdl ielrvmleda edrkypendl\n",
      "      781 frklrdavke aetcgsvaql llskkqkhrq ssdsgktrtk ltveelkafv qqlvslpcvi\n",
      "      841 sqtrqvknll ddveefhera qeammdetpd ssklqmlidm gsslyvelpe lprlkqelqq\n",
      "      901 arwldevrlt lsdpqqvtld vmkklidsgv glaphhavek amaelqellt vserweekak\n",
      "      961 vclqarprhs manlenivne aknipaflpn vlslkealqk arewtakvea iqsgnnyayl\n",
      "     1021 eqleslsakg rpipvrldal pqvesqvaaa rawrertgrt flkknsshtl lqvlsprtdi\n",
      "     1081 gvygsgknrr kkvkeiieke kekdldlepl sdleeglees rdtamvvavf kereqkeiea\n",
      "     1141 mhslraanla kmtiveriee vkfcicrkta sgfmlqcelc kdwfhnscvp lpksssqkkg\n",
      "     1201 sswqakdvkf lcplcmrsrr prletilsll vslqklpvrl pegealqclt eramswqdka\n",
      "     1261 rqalatdels salaklsvls qrmveqaare ktekiisael qkaaanpdlq ghlpsfqqsa\n",
      "     1321 fnrvvssvss sphqtmdydd eetdsdedir etygydmkdt asvkssssle pnlfcdeeip\n",
      "     1381 ikseevvthm wtapsfcaeh ayssaskscs qgsstprkqp rksplvprsl eppvlelspg\n",
      "     1441 akaqleelmm vgdllevsld etqhiwrilq athppsedrf lhimeddsie ekplkmkgkd\n",
      "     1501 ssekkrkrkl ekveqlfgeg kqktkelkki dkpkkkklkl nvdkskelnk lakklakeee\n",
      "     1561 rkkkkekaaa akvelvkest ekkrerkvld ipskydwsga eesddenavc aaqncqrpck\n",
      "     1621 dkvdwvqcdg gcdewfhqvc vgvsaemaen edyicincak kqgpdspgqa ppppflmsyk\n",
      "     1681 lpmedlkets\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#My KDM5s Protein list\n",
    "handle = Entrez.efetch(db=\"protein\", id=KDM5s[0], rettype=\"gb\", retmode=\"text\")\n",
    "print(handle.read())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The arguments rettype=\"gb\" and retmode=\"text\" let us download this record in the GenBank format.\n",
    "\n",
    "Note that until Easter 2009, the Entrez EFetch API let you use “genbank” as the return type, however the NCBI now insist on using the official return types of “gb” or “gbwithparts” (or “gp” for proteins) as described on online. Also note that until Feb 2012, the Entrez EFetch API would default to returning plain text files, but now defaults to XML.\n",
    "\n",
    "Alternatively, you could for example use rettype=\"fasta\" to get the Fasta-format; see the EFetch Sequences Help page for other options. Remember – the available formats depend on which database you are downloading from - see the main EFetch Help page.\n",
    "\n",
    "If you fetch the record in one of the formats accepted by Bio.SeqIO (see Chapter 5), you could directly parse it into a SeqRecord:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       KDM5A_MOUSE             1690 aa            linear   ROD 28-MAR-2018\n",
      "DEFINITION  RecName: Full=Lysine-specific demethylase 5A; AltName: Full=Histone\n",
      "            demethylase JARID1A; AltName: Full=Jumonji/ARID domain-containing\n",
      "            protein 1A; AltName: Full=Retinoblastoma-binding protein 2;\n",
      "            Short=RBBP-2.\n",
      "ACCESSION   Q3UXZ9\n",
      "VERSION     Q3UXZ9.2\n",
      "DBSOURCE    UniProtKB: locus KDM5A_MOUSE, accession Q3UXZ9;\n",
      "            class: standard.\n",
      "            extra accessions:Q3TM94,Q3UMI5,Q66JZ3\n",
      "            created: Jun 26, 2007.\n",
      "            sequence updated: Jun 26, 2007.\n",
      "            annotation updated: Mar 28, 2018.\n",
      "            xrefs: AC155720.7, AC078896.23, AK135085.1, BAE22414.1, AK144877.1,\n",
      "            BAE26113.1, AK166055.1, BAE38548.1, BC080691.1, AAH80691.1,\n",
      "            NP_666109.2\n",
      "            xrefs (non-sequence databases): CCDS:CCDS51889.1,\n",
      "            UniGene:Mm.404761, UniGene:Mm.463658, ProteinModelPortal:Q3UXZ9,\n",
      "            SMR:Q3UXZ9, BioGrid:229572, IntAct:Q3UXZ9,\n",
      "            STRING:10090.ENSMUSP00000098558, BindingDB:Q3UXZ9,\n",
      "            ChEMBL:CHEMBL3707461, iPTMnet:Q3UXZ9, PhosphoSitePlus:Q3UXZ9,\n",
      "            EPD:Q3UXZ9, MaxQB:Q3UXZ9, PaxDb:Q3UXZ9, PeptideAtlas:Q3UXZ9,\n",
      "            PRIDE:Q3UXZ9, Ensembl:ENSMUST00000005108,\n",
      "            Ensembl:ENSMUSP00000005108, Ensembl:ENSMUSG00000030180,\n",
      "            GeneID:214899, KEGG:mmu:214899, UCSC:uc009dne.2, CTD:5927,\n",
      "            MGI:2136980, eggNOG:KOG1246, eggNOG:ENOG410XR9J,\n",
      "            GeneTree:ENSGT00530000063118, HOGENOM:HOG000290719,\n",
      "            InParanoid:Q3UXZ9, KO:K11446, OMA:CHEWYHG, PhylomeDB:Q3UXZ9,\n",
      "            TreeFam:TF106476, BRENDA:1.14.11.B2, Reactome:R-MMU-3214842,\n",
      "            ChiTaRS:Kdm5a, PRO:PR:Q3UXZ9, Proteomes:UP000000589,\n",
      "            Bgee:ENSMUSG00000030180, CleanEx:MM_RBP2, ExpressionAtlas:Q3UXZ9,\n",
      "            Genevisible:Q3UXZ9, GO:0019907, GO:0005730, GO:0005634, GO:0032993,\n",
      "            GO:0003682, GO:0031490, GO:0001046, GO:0051213, GO:0003677,\n",
      "            GO:0003700, GO:0042393, GO:0034648, GO:0034647, GO:0035064,\n",
      "            GO:0003713, GO:0008270, GO:0032922, GO:0034720, GO:0008584,\n",
      "            GO:1901726, GO:0000122, GO:0045893, GO:0051090, GO:0007283,\n",
      "            GO:0006351, Gene3D:1.10.150.60, Gene3D:3.30.40.10,\n",
      "            InterPro:IPR001606, InterPro:IPR036431, InterPro:IPR003347,\n",
      "            InterPro:IPR003349, InterPro:IPR013637, InterPro:IPR019786,\n",
      "            InterPro:IPR004198, InterPro:IPR011011, InterPro:IPR001965,\n",
      "            InterPro:IPR019787, InterPro:IPR013083, Pfam:PF01388, Pfam:PF02373,\n",
      "            Pfam:PF02375, Pfam:PF00628, Pfam:PF08429, Pfam:PF02928,\n",
      "            SMART:SM00501, SMART:SM00558, SMART:SM00545, SMART:SM00249,\n",
      "            SUPFAM:SSF46774, SUPFAM:SSF57903, PROSITE:PS51011, PROSITE:PS51184,\n",
      "            PROSITE:PS51183, PROSITE:PS01359, PROSITE:PS50016\n",
      "KEYWORDS    Activator; Biological rhythms; Chromatin regulator; Complete\n",
      "            proteome; Developmental protein; Dioxygenase; Direct protein\n",
      "            sequencing; Iron; Isopeptide bond; Metal-binding; Nucleus;\n",
      "            Oxidoreductase; Phosphoprotein; Reference proteome; Repeat;\n",
      "            Transcription; Transcription regulation; Ubl conjugation; Zinc;\n",
      "            Zinc-finger.\n",
      "SOURCE      Mus musculus (house mouse)\n",
      "  ORGANISM  Mus musculus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Glires; Rodentia; Myomorpha;\n",
      "            Muroidea; Muridae; Murinae; Mus; Mus.\n",
      "REFERENCE   1  (residues 1 to 1690)\n",
      "  AUTHORS   Church,D.M., Goodstadt,L., Hillier,L.W., Zody,M.C., Goldstein,S.,\n",
      "            She,X., Bult,C.J., Agarwala,R., Cherry,J.L., DiCuccio,M.,\n",
      "            Hlavina,W., Kapustin,Y., Meric,P., Maglott,D., Birtle,Z.,\n",
      "            Marques,A.C., Graves,T., Zhou,S., Teague,B., Potamousis,K.,\n",
      "            Churas,C., Place,M., Herschleb,J., Runnheim,R., Forrest,D.,\n",
      "            Amos-Landgraf,J., Schwartz,D.C., Cheng,Z., Lindblad-Toh,K.,\n",
      "            Eichler,E.E. and Ponting,C.P.\n",
      "  CONSRTM   Mouse Genome Sequencing Consortium\n",
      "  TITLE     Lineage-specific biology revealed by a finished genome assembly of\n",
      "            the mouse\n",
      "  JOURNAL   PLoS Biol. 7 (5), e1000112 (2009)\n",
      "   PUBMED   19468303\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA].;\n",
      "            STRAIN=C57BL/6J\n",
      "REFERENCE   2  (residues 1 to 1690)\n",
      "  AUTHORS   Carninci,P., Kasukawa,T., Katayama,S., Gough,J., Frith,M.C.,\n",
      "            Maeda,N., Oyama,R., Ravasi,T., Lenhard,B., Wells,C., Kodzius,R.,\n",
      "            Shimokawa,K., Bajic,V.B., Brenner,S.E., Batalov,S., Forrest,A.R.,\n",
      "            Zavolan,M., Davis,M.J., Wilming,L.G., Aidinis,V., Allen,J.E.,\n",
      "            Ambesi-Impiombato,A., Apweiler,R., Aturaliya,R.N., Bailey,T.L.,\n",
      "            Bansal,M., Baxter,L., Beisel,K.W., Bersano,T., Bono,H., Chalk,A.M.,\n",
      "            Chiu,K.P., Choudhary,V., Christoffels,A., Clutterbuck,D.R.,\n",
      "            Crowe,M.L., Dalla,E., Dalrymple,B.P., de Bono,B., Della Gatta,G.,\n",
      "            di Bernardo,D., Down,T., Engstrom,P., Fagiolini,M., Faulkner,G.,\n",
      "            Fletcher,C.F., Fukushima,T., Furuno,M., Futaki,S., Gariboldi,M.,\n",
      "            Georgii-Hemming,P., Gingeras,T.R., Gojobori,T., Green,R.E.,\n",
      "            Gustincich,S., Harbers,M., Hayashi,Y., Hensch,T.K., Hirokawa,N.,\n",
      "            Hill,D., Huminiecki,L., Iacono,M., Ikeo,K., Iwama,A., Ishikawa,T.,\n",
      "            Jakt,M., Kanapin,A., Katoh,M., Kawasawa,Y., Kelso,J., Kitamura,H.,\n",
      "            Kitano,H., Kollias,G., Krishnan,S.P., Kruger,A., Kummerfeld,S.K.,\n",
      "            Kurochkin,I.V., Lareau,L.F., Lazarevic,D., Lipovich,L., Liu,J.,\n",
      "            Liuni,S., McWilliam,S., Madan Babu,M., Madera,M., Marchionni,L.,\n",
      "            Matsuda,H., Matsuzawa,S., Miki,H., Mignone,F., Miyake,S.,\n",
      "            Morris,K., Mottagui-Tabar,S., Mulder,N., Nakano,N., Nakauchi,H.,\n",
      "            Ng,P., Nilsson,R., Nishiguchi,S., Nishikawa,S., Nori,F., Ohara,O.,\n",
      "            Okazaki,Y., Orlando,V., Pang,K.C., Pavan,W.J., Pavesi,G.,\n",
      "            Pesole,G., Petrovsky,N., Piazza,S., Reed,J., Reid,J.F., Ring,B.Z.,\n",
      "            Ringwald,M., Rost,B., Ruan,Y., Salzberg,S.L., Sandelin,A.,\n",
      "            Schneider,C., Schonbach,C., Sekiguchi,K., Semple,C.A., Seno,S.,\n",
      "            Sessa,L., Sheng,Y., Shibata,Y., Shimada,H., Shimada,K., Silva,D.,\n",
      "            Sinclair,B., Sperling,S., Stupka,E., Sugiura,K., Sultana,R.,\n",
      "            Takenaka,Y., Taki,K., Tammoja,K., Tan,S.L., Tang,S., Taylor,M.S.,\n",
      "            Tegner,J., Teichmann,S.A., Ueda,H.R., van Nimwegen,E., Verardo,R.,\n",
      "            Wei,C.L., Yagi,K., Yamanishi,H., Zabarovsky,E., Zhu,S., Zimmer,A.,\n",
      "            Hide,W., Bult,C., Grimmond,S.M., Teasdale,R.D., Liu,E.T.,\n",
      "            Brusic,V., Quackenbush,J., Wahlestedt,C., Mattick,J.S., Hume,D.A.,\n",
      "            Kai,C., Sasaki,D., Tomaru,Y., Fukuda,S., Kanamori-Katayama,M.,\n",
      "            Suzuki,M., Aoki,J., Arakawa,T., Iida,J., Imamura,K., Itoh,M.,\n",
      "            Kato,T., Kawaji,H., Kawagashira,N., Kawashima,T., Kojima,M.,\n",
      "            Kondo,S., Konno,H., Nakano,K., Ninomiya,N., Nishio,T., Okada,M.,\n",
      "            Plessy,C., Shibata,K., Shiraki,T., Suzuki,S., Tagami,M., Waki,K.,\n",
      "            Watahiki,A., Okamura-Oho,Y., Suzuki,H., Kawai,J. and Hayashizaki,Y.\n",
      "  CONSRTM   FANTOM Consortium; RIKEN Genome Exploration Research Group and\n",
      "            Genome Science Group (Genome Network Project Core Group)\n",
      "  TITLE     The transcriptional landscape of the mammalian genome\n",
      "  JOURNAL   Science 309 (5740), 1559-1563 (2005)\n",
      "   PUBMED   16141072\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] OF 1-1552.;\n",
      "            STRAIN=C57BL/6J; TISSUE=Lung, and Olfactory bulb\n",
      "            Erratum:[Science. 2006 Mar 24;311(5768):1713]\n",
      "REFERENCE   3  (residues 1 to 1690)\n",
      "  AUTHORS   Gerhard,D.S., Wagner,L., Feingold,E.A., Shenmen,C.M., Grouse,L.H.,\n",
      "            Schuler,G., Klein,S.L., Old,S., Rasooly,R., Good,P., Guyer,M.,\n",
      "            Peck,A.M., Derge,J.G., Lipman,D., Collins,F.S., Jang,W., Sherry,S.,\n",
      "            Feolo,M., Misquitta,L., Lee,E., Rotmistrovsky,K., Greenhut,S.F.,\n",
      "            Schaefer,C.F., Buetow,K., Bonner,T.I., Haussler,D., Kent,J.,\n",
      "            Kiekhaus,M., Furey,T., Brent,M., Prange,C., Schreiber,K.,\n",
      "            Shapiro,N., Bhat,N.K., Hopkins,R.F., Hsie,F., Driscoll,T.,\n",
      "            Soares,M.B., Casavant,T.L., Scheetz,T.E., Brown-stein,M.J.,\n",
      "            Usdin,T.B., Toshiyuki,S., Carninci,P., Piao,Y., Dudekula,D.B.,\n",
      "            Ko,M.S., Kawakami,K., Suzuki,Y., Sugano,S., Gruber,C.E.,\n",
      "            Smith,M.R., Simmons,B., Moore,T., Waterman,R., Johnson,S.L.,\n",
      "            Ruan,Y., Wei,C.L., Mathavan,S., Gunaratne,P.H., Wu,J., Garcia,A.M.,\n",
      "            Hulyk,S.W., Fuh,E., Yuan,Y., Sneed,A., Kowis,C., Hodgson,A.,\n",
      "            Muzny,D.M., McPherson,J., Gibbs,R.A., Fahey,J., Helton,E.,\n",
      "            Ketteman,M., Madan,A., Rodrigues,S., Sanchez,A., Whiting,M.,\n",
      "            Madari,A., Young,A.C., Wetherby,K.D., Granite,S.J., Kwong,P.N.,\n",
      "            Brinkley,C.P., Pearson,R.L., Bouffard,G.G., Blakesly,R.W.,\n",
      "            Green,E.D., Dickson,M.C., Rodriguez,A.C., Grimwood,J., Schmutz,J.,\n",
      "            Myers,R.M., Butterfield,Y.S., Griffith,M., Griffith,O.L.,\n",
      "            Krzywinski,M.I., Liao,N., Morin,R., Palmquist,D., Petrescu,A.S.,\n",
      "            Skalska,U., Smailus,D.E., Stott,J.M., Schnerch,A., Schein,J.E.,\n",
      "            Jones,S.J., Holt,R.A., Baross,A., Marra,M.A., Clifton,S.,\n",
      "            Makowski,K.A., Bosak,S. and Malek,J.\n",
      "  CONSRTM   MGC Project Team\n",
      "  TITLE     The status, quality, and expansion of the NIH full-length cDNA\n",
      "            project: the Mammalian Gene Collection (MGC)\n",
      "  JOURNAL   Genome Res. 14 (10B), 2121-2127 (2004)\n",
      "   PUBMED   15489334\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] OF 1-1099.;\n",
      "            STRAIN=C57BL/6J; TISSUE=Germ cell\n",
      "            Erratum:[Genome Res. 2006 Jun;16(6):804. Morrin, Ryan [corrected to\n",
      "            Morin, Ryan]]\n",
      "REFERENCE   4  (residues 1 to 1690)\n",
      "  AUTHORS   Lubec,G., Sunyer,B. and Chen,W.-Q.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-JAN-2009) to UniProtKB\n",
      "  REMARK    PROTEIN SEQUENCE OF 1538-1546, AND IDENTIFICATION BY MASS\n",
      "            SPECTROMETRY.;\n",
      "            STRAIN=OF1; TISSUE=Hippocampus\n",
      "REFERENCE   5  (residues 1 to 1690)\n",
      "  AUTHORS   Klose,R.J., Yan,Q., Tothova,Z., Yamane,K., Erdjument-Bromage,H.,\n",
      "            Tempst,P., Gilliland,D.G., Zhang,Y. and Kaelin,W.G. Jr.\n",
      "  TITLE     The retinoblastoma binding protein RBP2 is an H3K4 demethylase\n",
      "  JOURNAL   Cell 128 (5), 889-900 (2007)\n",
      "   PUBMED   17320163\n",
      "  REMARK    FUNCTION, AND DISRUPTION PHENOTYPE.\n",
      "REFERENCE   6  (residues 1 to 1690)\n",
      "  AUTHORS   Christensen,J., Agger,K., Cloos,P.A., Pasini,D., Rose,S.,\n",
      "            Sennels,L., Rappsilber,J., Hansen,K.H., Salcini,A.E. and Helin,K.\n",
      "  TITLE     RBP2 belongs to a family of demethylases, specific for tri-and\n",
      "            dimethylated lysine 4 on histone 3\n",
      "  JOURNAL   Cell 128 (6), 1063-1076 (2007)\n",
      "   PUBMED   17320161\n",
      "  REMARK    FUNCTION.\n",
      "REFERENCE   7  (residues 1 to 1690)\n",
      "  AUTHORS   Peng,J.C., Valouev,A., Swigut,T., Zhang,J., Zhao,Y., Sidow,A. and\n",
      "            Wysocka,J.\n",
      "  TITLE     Jarid2/Jumonji coordinates control of PRC2 enzymatic activity and\n",
      "            target gene occupancy in pluripotent cells\n",
      "  JOURNAL   Cell 139 (7), 1290-1302 (2009)\n",
      "   PUBMED   20064375\n",
      "  REMARK    SUBCELLULAR LOCATION, DOMAIN GSGFP MOTIF, AND INTERACTION WITH\n",
      "            SUZ12.\n",
      "REFERENCE   8  (residues 1 to 1690)\n",
      "  AUTHORS   Trost,M., English,L., Lemieux,S., Courcelles,M., Desjardins,M. and\n",
      "            Thibault,P.\n",
      "  TITLE     The phagosomal proteome in interferon-gamma-activated macrophages\n",
      "  JOURNAL   Immunity 30 (1), 143-154 (2009)\n",
      "   PUBMED   19144319\n",
      "  REMARK    PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-1111, AND\n",
      "            IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].\n",
      "REFERENCE   9  (residues 1 to 1690)\n",
      "  AUTHORS   Huttlin,E.L., Jedrychowski,M.P., Elias,J.E., Goswami,T., Rad,R.,\n",
      "            Beausoleil,S.A., Villen,J., Haas,W., Sowa,M.E. and Gygi,S.P.\n",
      "  TITLE     A tissue-specific atlas of mouse protein phosphorylation and\n",
      "            expression\n",
      "  JOURNAL   Cell 143 (7), 1174-1189 (2010)\n",
      "   PUBMED   21183079\n",
      "  REMARK    PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-1111; THR-1343;\n",
      "            SER-1345; TYR-1595; SER-1598; SER-1603 AND SER-1666, AND\n",
      "            IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS].;\n",
      "            TISSUE=Brain, Kidney, Lung, Spleen, and Testis\n",
      "REFERENCE   10 (residues 1 to 1690)\n",
      "  AUTHORS   DiTacchio,L., Le,H.D., Vollmers,C., Hatori,M., Witcher,M.,\n",
      "            Secombe,J. and Panda,S.\n",
      "  TITLE     Histone lysine demethylase JARID1a activates CLOCK-BMAL1 and\n",
      "            influences the circadian clock\n",
      "  JOURNAL   Science 333 (6051), 1881-1885 (2011)\n",
      "   PUBMED   21960634\n",
      "  REMARK    FUNCTION, AND INTERACTION WITH ARNTL; CLOCK AND HDAC1.\n",
      "COMMENT     On or before Jul 3, 2007 this sequence version replaced\n",
      "            gi:123783469, gi:123784837, gi:81884291, gi:123795350.\n",
      "            [FUNCTION] Histone demethylase that specifically demethylates\n",
      "            'Lys-4' of histone H3, thereby playing a central role in histone\n",
      "            code. Does not demethylate histone H3 'Lys-9', H3 'Lys-27', H3\n",
      "            'Lys-36', H3 'Lys-79' or H4 'Lys-20'. Demethylates trimethylated\n",
      "            and dimethylated but not monomethylated H3 'Lys-4'\n",
      "            (PubMed:17320161, PubMed:17320163). Regulates specific gene\n",
      "            transcription through DNA-binding on 5'-CCGCCC-3' motif. May\n",
      "            stimulate transcription mediated by nuclear receptors. Involved in\n",
      "            transcriptional regulation of Hox proteins during cell\n",
      "            differentiation (By similarity). May participate in transcriptional\n",
      "            repression of cytokines such as CXCL12. Plays a role in the\n",
      "            regulation of the circadian rhythm and in maintaining the normal\n",
      "            periodicity of the circadian clock. In a histone\n",
      "            demethylase-independent manner, acts as a coactivator of the\n",
      "            CLOCK-ARNTL/BMAL1-mediated transcriptional activation of PER1/2 and\n",
      "            other clock-controlled genes and increases histone acetylation at\n",
      "            PER1/2 promoters by inhibiting the activity of HDAC1\n",
      "            (PubMed:21960634). Seems to act as a transcriptional corepressor\n",
      "            for some genes such as MT1F and to favor the proliferation of\n",
      "            cancer cells (By similarity). {ECO:0000250|UniProtKB:P29375,\n",
      "            ECO:0000269|PubMed:17320161, ECO:0000269|PubMed:17320163,\n",
      "            ECO:0000269|PubMed:21960634}.\n",
      "            [COFACTOR] Name=Fe(2+); Xref=ChEBI:CHEBI:29033;\n",
      "            Evidence={ECO:0000250}; Note=Binds 1 Fe(2+) ion per subunit.\n",
      "            {ECO:0000250}.\n",
      "            [SUBUNIT] Interacts with RB1, ESR1, MYC, MYCN and LMO2 (By\n",
      "            similarity). Interacts with SUZ12; the interaction is direct\n",
      "            (PubMed:20064375). Interacts with HDAC1, ARNTL/BMAL1 and CLOCK\n",
      "            (PubMed:21960634). Interacts (via PHD-type 1 zinc finger) with\n",
      "            histone H3 unmodified at 'Lys-4' and (via PHD-type 3 zinc finger)\n",
      "            with histone H3 di- and trimethylated at 'Lys-4' (By similarity).\n",
      "            {ECO:0000250|UniProtKB:P29375, ECO:0000269|PubMed:20064375,\n",
      "            ECO:0000269|PubMed:21960634}.\n",
      "            [INTERACTION] Q80U70:Suz12; NbExp=2; IntAct=EBI-2531441,\n",
      "            EBI-2526494.\n",
      "            [SUBCELLULAR LOCATION] Nucleus, nucleolus\n",
      "            {ECO:0000250|UniProtKB:P29375}. Nucleus\n",
      "            {ECO:0000255|PROSITE-ProRule:PRU00355,\n",
      "            ECO:0000255|PROSITE-ProRule:PRU00537, ECO:0000269|PubMed:20064375}.\n",
      "            Note=Occupies promoters of genes involved in RNA metabolism and\n",
      "            mitochondrial function. {ECO:0000269|PubMed:20064375}.\n",
      "            [DOMAIN] The GSGFP motif is required for the interaction with SUZ12\n",
      "            (PubMed:20064375). The ARID domain specifically binds to the CCGCCC\n",
      "            motif and is required for the lysine-specific histone demethylase\n",
      "            activity. The PHD-type 3 zinc finger is required for the\n",
      "            interaction with histone H3 di- and trimethylated at 'Lys-4' (By\n",
      "            similarity). {ECO:0000250|UniProtKB:P29375,\n",
      "            ECO:0000269|PubMed:20064375}.\n",
      "            [DISRUPTION PHENOTYPE] Mice are grossly normal, except that they\n",
      "            exhibit behavioral abnormalities when held upside down by the tail,\n",
      "            and slight hematological defects. {ECO:0000269|PubMed:17320163}.\n",
      "            [SIMILARITY] Belongs to the JARID1 histone demethylase family.\n",
      "            {ECO:0000305}.\n",
      "            [SEQUENCE CAUTION] Sequence=BAE22414.1; Type=Erroneous initiation;\n",
      "            Note=Translation N-terminally extended.; Evidence={ECO:0000305}.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..1690\n",
      "                     /organism=\"Mus musculus\"\n",
      "                     /db_xref=\"taxon:10090\"\n",
      "     gene            1..1690\n",
      "                     /gene=\"Kdm5a\"\n",
      "                     /gene_synonym=\"Jarid1a\"\n",
      "                     /gene_synonym=\"Rbp2\"\n",
      "     Protein         1..1690\n",
      "                     /product=\"Lysine-specific demethylase 5A\"\n",
      "                     /EC_number=\"1.14.11.-\"\n",
      "                     /note=\"Histone demethylase JARID1A; Jumonji/ARID\n",
      "                     domain-containing protein 1A; Retinoblastoma-binding\n",
      "                     protein 2; RBBP-2\"\n",
      "                     /UniProtKB_evidence=\"Evidence at protein level\"\n",
      "     Region          1..1690\n",
      "                     /region_name=\"Mature chain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Lysine-specific demethylase 5A.\n",
      "                     /FTId=PRO_0000292411.\"\n",
      "     Region          10\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"A -> S (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          14\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"V -> R (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          17\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"P -> A (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          18..59\n",
      "                     /region_name=\"JmjN\"\n",
      "                     /note=\"Small domain found in the jumonji family of\n",
      "                     transcription factors; smart00545\"\n",
      "                     /db_xref=\"CDD:128818\"\n",
      "     Region          19..60\n",
      "                     /region_name=\"Domain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"JmjN. {ECO:0000255|PROSITE-ProRule:PRU00537}.\"\n",
      "     Region          24\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"P -> A (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Region          84..174\n",
      "                     /region_name=\"Domain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"ARID. {ECO:0000255|PROSITE-ProRule:PRU00355}.\"\n",
      "     Region          85..170\n",
      "                     /region_name=\"ARID\"\n",
      "                     /note=\"ARID/BRIGHT DNA binding domain; smart01014\"\n",
      "                     /db_xref=\"CDD:198082\"\n",
      "     Bond            bond(191)\n",
      "                     /bond_type=\"xlink\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
      "                     G-Cter in SUMO2). {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            204\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          293..343\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"PHD-type 1.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00146}.\"\n",
      "     Region          295..343\n",
      "                     /region_name=\"PHD1_KDM5A\"\n",
      "                     /note=\"PHD finger 1 found in Lysine-specific demethylase\n",
      "                     5A (KDM5A); cd15602\"\n",
      "                     /db_xref=\"CDD:277075\"\n",
      "     Site            order(295,307..311,315,335)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"putative histone H3 binding site [polypeptide\n",
      "                     binding]\"\n",
      "                     /db_xref=\"CDD:277075\"\n",
      "     Site            order(296,299,311,314,319,322,337,340)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"Zn binding site [ion binding]\"\n",
      "                     /db_xref=\"CDD:277075\"\n",
      "     Site            409\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          419..423\n",
      "                     /region_name=\"Short sequence motif of biological interest\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"GSGFP motif. {ECO:0000269|PubMed:20064375}.\"\n",
      "     Region          437..603\n",
      "                     /region_name=\"Domain\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"JmjC. {ECO:0000255|PROSITE-ProRule:PRU00538}.\"\n",
      "     Region          470..586\n",
      "                     /region_name=\"JmjC\"\n",
      "                     /note=\"JmjC domain, hydroxylase; pfam02373\"\n",
      "                     /db_xref=\"CDD:202224\"\n",
      "     Site            483\n",
      "                     /site_type=\"metal-binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Iron; catalytic.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00538}.\"\n",
      "     Site            485\n",
      "                     /site_type=\"metal-binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Iron; catalytic. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            491\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            493\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            501\n",
      "                     /site_type=\"binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"2-oxoglutarate. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          544\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"V -> A (in Ref. 2; BAE38548). {ECO:0000305}.\"\n",
      "     Site            571\n",
      "                     /site_type=\"metal-binding\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Iron; catalytic.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00538}.\"\n",
      "     Region          676..728\n",
      "                     /region_name=\"zf-C5HC2\"\n",
      "                     /note=\"C5HC2 zinc finger; pfam02928\"\n",
      "                     /db_xref=\"CDD:280996\"\n",
      "     Region          676..728\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"C5HC2. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          741..1070\n",
      "                     /region_name=\"PLU-1\"\n",
      "                     /note=\"PLU-1-like protein; pfam08429\"\n",
      "                     /db_xref=\"CDD:285609\"\n",
      "     Bond            bond(1007)\n",
      "                     /bond_type=\"xlink\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Glycyl lysine isopeptide (Lys-Gly) (interchain with\n",
      "                     G-Cter in SUMO2). {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1111\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:19144319,\n",
      "                     ECO:0000244|PubMed:21183079}.\"\n",
      "     Region          1153..1210\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"PHD-type 2.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00146}.\"\n",
      "     Region          1163..1215\n",
      "                     /region_name=\"PHD2_KDM5A\"\n",
      "                     /note=\"PHD finger 2 found in Lysine-specific demethylase\n",
      "                     5A (KDM5A); cd15606\"\n",
      "                     /db_xref=\"CDD:277079\"\n",
      "     Site            order(1163,1173..1177,1181,1210)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"putative histone H3 binding site [polypeptide\n",
      "                     binding]\"\n",
      "                     /db_xref=\"CDD:277079\"\n",
      "     Site            order(1164,1166,1177,1180,1185,1188,1212,1215)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"Zn binding site [ion binding]\"\n",
      "                     /db_xref=\"CDD:277079\"\n",
      "     Region          1272\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"A -> P (in Ref. 2; BAE22414). {ECO:0000305}.\"\n",
      "     Site            1330\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1331\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1343\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphothreonine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Site            1345\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Site            1438\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Region          1484..1579\n",
      "                     /region_name=\"Compositionally biased region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Lys-rich.\"\n",
      "     Site            1488\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            1595\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphotyrosine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Site            1598\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Region          1599..1653\n",
      "                     /region_name=\"Zinc finger region\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"PHD-type 3.\n",
      "                     {ECO:0000255|PROSITE-ProRule:PRU00146}.\"\n",
      "     Site            1603\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "     Region          1608..1659\n",
      "                     /region_name=\"PHD3_KDM5A\"\n",
      "                     /note=\"PHD finger 3 found in Lysine-specific demethylase\n",
      "                     5A (KDM5A); cd15686\"\n",
      "                     /db_xref=\"CDD:277156\"\n",
      "     Site            order(1610,1615,1628,1632,1637,1640,1655,1658)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"Zn binding site [ion binding]\"\n",
      "                     /db_xref=\"CDD:277156\"\n",
      "     Region          1622..1690\n",
      "                     /region_name=\"Region of interest in the sequence\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Interaction with LMO2.\n",
      "                     {ECO:0000250|UniProtKB:P29375}.\"\n",
      "     Site            order(1624..1629,1633,1635,1638)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"H3K4me3 binding site [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:277156\"\n",
      "     Site            1666\n",
      "                     /site_type=\"modified\"\n",
      "                     /experiment=\"experimental evidence, no additional details\n",
      "                     recorded\"\n",
      "                     /note=\"Phosphoserine. {ECO:0000244|PubMed:21183079}.\"\n",
      "ORIGIN      \n",
      "        1 masvgpggya aefvpppecp vfepsweeft dplsfigrir pfaektgick irppkdwqpp\n",
      "       61 facevktfrf tprvqrlnel eamtrvrldf ldqlakfwel qgstlkipvv erkildlyal\n",
      "      121 skivaskggf eivtkekkws kvgsrlgylp gkgtgsllks hyerilypye lfqsgvslmg\n",
      "      181 vqmpdldlke kveaevlstd iqpspergtr mnippkrtrr vksqsdsgev nrntelkklq\n",
      "      241 ifgagpkvvg lavgakdked evtrrrkvtn rsdafnmqmr qrkgtlsvnf vdlyvcmfcg\n",
      "      301 rgnnedklll cdgcddsyht fcllpplpdv pkgdwrcpkc vaeecnkpre afgfeqavre\n",
      "      361 ytlqsfgema dnfksdyfnm pvhmvptelv ekefwrlvss ieedviveyg adisskdfgs\n",
      "      421 gfpkkdgqrk mlpeeeeyal sgwnlnnmpv leqsvlahin vdisgmkvpw lyvgmcfssf\n",
      "      481 cwhiedhwsy sinylhwgep ktwygvpsha aeqleevmre lapelfesqp dllhqlvtim\n",
      "      541 npnvlmehgv pvyrtnqcag efvvtfpray hsgfnqgynf aeavnfctad wlpigrqcvn\n",
      "      601 hyrrlrrhcv fsheelifkm aadpecldvg laamvckelt lmteeetrlr esvvqmgvvm\n",
      "      661 seeevfelvp dderqcsacr ttcflsaltc scnperlvcl yhptdlcscp mqnkclryry\n",
      "      721 pledlpslly gvkvraqsyd twvnrvteal sasfnhkkdl ielrvmleda edrkypendl\n",
      "      781 frklrdavke aetcgsvaql llskkqkhrq ssdsgktrtk ltveelkafv qqlvslpcvi\n",
      "      841 sqtrqvknll ddveefhera qeammdetpd ssklqmlidm gsslyvelpe lprlkqelqq\n",
      "      901 arwldevrlt lsdpqqvtld vmkklidsgv glaphhavek amaelqellt vserweekak\n",
      "      961 vclqarprhs manlenivne aknipaflpn vlslkealqk arewtakvea iqsgnnyayl\n",
      "     1021 eqleslsakg rpipvrldal pqvesqvaaa rawrertgrt flkknsshtl lqvlsprtdi\n",
      "     1081 gvygsgknrr kkvkeiieke kekdldlepl sdleeglees rdtamvvavf kereqkeiea\n",
      "     1141 mhslraanla kmtiveriee vkfcicrkta sgfmlqcelc kdwfhnscvp lpksssqkkg\n",
      "     1201 sswqakdvkf lcplcmrsrr prletilsll vslqklpvrl pegealqclt eramswqdka\n",
      "     1261 rqalatdels salaklsvls qrmveqaare ktekiisael qkaaanpdlq ghlpsfqqsa\n",
      "     1321 fnrvvssvss sphqtmdydd eetdsdedir etygydmkdt asvkssssle pnlfcdeeip\n",
      "     1381 ikseevvthm wtapsfcaeh ayssaskscs qgsstprkqp rksplvprsl eppvlelspg\n",
      "     1441 akaqleelmm vgdllevsld etqhiwrilq athppsedrf lhimeddsie ekplkmkgkd\n",
      "     1501 ssekkrkrkl ekveqlfgeg kqktkelkki dkpkkkklkl nvdkskelnk lakklakeee\n",
      "     1561 rkkkkekaaa akvelvkest ekkrerkvld ipskydwsga eesddenavc aaqncqrpck\n",
      "     1621 dkvdwvqcdg gcdewfhqvc vgvsaemaen edyicincak kqgpdspgqa ppppflmsyk\n",
      "     1681 lpmedlkets\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Try changing to 'gp' = No obvious difference\n",
    "#My KDM5s Protein list\n",
    "handle = Entrez.efetch(db=\"protein\", id=KDM5s[0], rettype=\"gp\", retmode=\"text\")\n",
    "print(handle.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp|Q3UXZ9.2|KDM5A_MOUSE RecName: Full=Lysine-specific demethylase 5A; AltName: Full=Histone demethylase JARID1A; AltName: Full=Jumonji/ARID domain-containing protein 1A; AltName: Full=Retinoblastoma-binding protein 2; Short=RBBP-2\n",
      "MASVGPGGYAAEFVPPPECPVFEPSWEEFTDPLSFIGRIRPFAEKTGICKIRPPKDWQPPFACEVKTFRF\n",
      "TPRVQRLNELEAMTRVRLDFLDQLAKFWELQGSTLKIPVVERKILDLYALSKIVASKGGFEIVTKEKKWS\n",
      "KVGSRLGYLPGKGTGSLLKSHYERILYPYELFQSGVSLMGVQMPDLDLKEKVEAEVLSTDIQPSPERGTR\n",
      "MNIPPKRTRRVKSQSDSGEVNRNTELKKLQIFGAGPKVVGLAVGAKDKEDEVTRRRKVTNRSDAFNMQMR\n",
      "QRKGTLSVNFVDLYVCMFCGRGNNEDKLLLCDGCDDSYHTFCLLPPLPDVPKGDWRCPKCVAEECNKPRE\n",
      "AFGFEQAVREYTLQSFGEMADNFKSDYFNMPVHMVPTELVEKEFWRLVSSIEEDVIVEYGADISSKDFGS\n",
      "GFPKKDGQRKMLPEEEEYALSGWNLNNMPVLEQSVLAHINVDISGMKVPWLYVGMCFSSFCWHIEDHWSY\n",
      "SINYLHWGEPKTWYGVPSHAAEQLEEVMRELAPELFESQPDLLHQLVTIMNPNVLMEHGVPVYRTNQCAG\n",
      "EFVVTFPRAYHSGFNQGYNFAEAVNFCTADWLPIGRQCVNHYRRLRRHCVFSHEELIFKMAADPECLDVG\n",
      "LAAMVCKELTLMTEEETRLRESVVQMGVVMSEEEVFELVPDDERQCSACRTTCFLSALTCSCNPERLVCL\n",
      "YHPTDLCSCPMQNKCLRYRYPLEDLPSLLYGVKVRAQSYDTWVNRVTEALSASFNHKKDLIELRVMLEDA\n",
      "EDRKYPENDLFRKLRDAVKEAETCGSVAQLLLSKKQKHRQSSDSGKTRTKLTVEELKAFVQQLVSLPCVI\n",
      "SQTRQVKNLLDDVEEFHERAQEAMMDETPDSSKLQMLIDMGSSLYVELPELPRLKQELQQARWLDEVRLT\n",
      "LSDPQQVTLDVMKKLIDSGVGLAPHHAVEKAMAELQELLTVSERWEEKAKVCLQARPRHSMANLENIVNE\n",
      "AKNIPAFLPNVLSLKEALQKAREWTAKVEAIQSGNNYAYLEQLESLSAKGRPIPVRLDALPQVESQVAAA\n",
      "RAWRERTGRTFLKKNSSHTLLQVLSPRTDIGVYGSGKNRRKKVKEIIEKEKEKDLDLEPLSDLEEGLEES\n",
      "RDTAMVVAVFKEREQKEIEAMHSLRAANLAKMTIVERIEEVKFCICRKTASGFMLQCELCKDWFHNSCVP\n",
      "LPKSSSQKKGSSWQAKDVKFLCPLCMRSRRPRLETILSLLVSLQKLPVRLPEGEALQCLTERAMSWQDKA\n",
      "RQALATDELSSALAKLSVLSQRMVEQAAREKTEKIISAELQKAAANPDLQGHLPSFQQSAFNRVVSSVSS\n",
      "SPHQTMDYDDEETDSDEDIRETYGYDMKDTASVKSSSSLEPNLFCDEEIPIKSEEVVTHMWTAPSFCAEH\n",
      "AYSSASKSCSQGSSTPRKQPRKSPLVPRSLEPPVLELSPGAKAQLEELMMVGDLLEVSLDETQHIWRILQ\n",
      "ATHPPSEDRFLHIMEDDSIEEKPLKMKGKDSSEKKRKRKLEKVEQLFGEGKQKTKELKKIDKPKKKKLKL\n",
      "NVDKSKELNKLAKKLAKEEERKKKKEKAAAAKVELVKESTEKKRERKVLDIPSKYDWSGAEESDDENAVC\n",
      "AAQNCQRPCKDKVDWVQCDGGCDEWFHQVCVGVSAEMAENEDYICINCAKKQGPDSPGQAPPPPFLMSYK\n",
      "LPMEDLKETS\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Try rettype=\"fasta\" = Less informative but more concise.\n",
    "#My KDM5s Protein list\n",
    "handle = Entrez.efetch(db=\"protein\", id=KDM5s[0], rettype=\"fasta\", retmode=\"text\")\n",
    "print(handle.read())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you fetch the record in one of the formats accepted by Bio.SeqIO (see Chapter 5), you could directly parse it into a SeqRecord:\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EU490707.1\n",
      "EU490707\n",
      "Selenipedium aequinoctiale maturase K (matK) gene, partial cds; chloroplast\n",
      "3\n",
      "Seq('ATTTTTTACGAACCTGTGGAAATTTTTGGTTATGACAATAAATCTAGTTTAGTA...GAA', IUPACAmbiguousDNA())\n"
     ]
    }
   ],
   "source": [
    "#Cookbook example\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id=\"EU490707\", rettype=\"gb\", retmode=\"text\")\n",
    "record = SeqIO.read(handle, \"genbank\")\n",
    "handle.close()\n",
    "print(record.id)\n",
    "print(record.name)\n",
    "print(record.description)\n",
    "print(len(record.features))\n",
    "print(repr(record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q3UXZ9.2\n",
      "KDM5A_MOUSE\n",
      "RecName: Full=Lysine-specific demethylase 5A; AltName: Full=Histone demethylase JARID1A; AltName: Full=Jumonji/ARID domain-containing protein 1A; AltName: Full=Retinoblastoma-binding protein 2; Short=RBBP-2\n",
      "55\n",
      "Seq('MASVGPGGYAAEFVPPPECPVFEPSWEEFTDPLSFIGRIRPFAEKTGICKIRPP...ETS', IUPACProtein())\n"
     ]
    }
   ],
   "source": [
    "#My version\n",
    "handle = Entrez.efetch(db=\"protein\", id=KDM5s[0], rettype=\"gb\", retmode=\"text\")\n",
    "record = SeqIO.read(handle, \"genbank\")\n",
    "handle.close()\n",
    "print(record.id)\n",
    "print(record.name)\n",
    "print(record.description)\n",
    "print(len(record.features))\n",
    "print(repr(record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(1690)), type='source'), SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(1690)), type='gene'), SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(1690)), type='Protein'), SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(1690)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(9), ExactPosition(10)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(13), ExactPosition(14)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(16), ExactPosition(17)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(17), ExactPosition(59)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(18), ExactPosition(60)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(23), ExactPosition(24)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(83), ExactPosition(174)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(84), ExactPosition(170)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(190), ExactPosition(191)), type='Bond'), SeqFeature(FeatureLocation(ExactPosition(203), ExactPosition(204)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(292), ExactPosition(343)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(294), ExactPosition(343)), type='Region'), SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(294), ExactPosition(295)), FeatureLocation(ExactPosition(306), ExactPosition(311)), FeatureLocation(ExactPosition(314), ExactPosition(315)), FeatureLocation(ExactPosition(334), ExactPosition(335))], 'order'), type='Site', location_operator='order'), SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(295), ExactPosition(296)), FeatureLocation(ExactPosition(298), ExactPosition(299)), FeatureLocation(ExactPosition(310), ExactPosition(311)), FeatureLocation(ExactPosition(313), ExactPosition(314)), FeatureLocation(ExactPosition(318), ExactPosition(319)), FeatureLocation(ExactPosition(321), ExactPosition(322)), FeatureLocation(ExactPosition(336), ExactPosition(337)), FeatureLocation(ExactPosition(339), ExactPosition(340))], 'order'), type='Site', location_operator='order'), SeqFeature(FeatureLocation(ExactPosition(408), ExactPosition(409)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(418), ExactPosition(423)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(436), ExactPosition(603)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(469), ExactPosition(586)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(482), ExactPosition(483)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(484), ExactPosition(485)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(490), ExactPosition(491)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(492), ExactPosition(493)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(500), ExactPosition(501)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(543), ExactPosition(544)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(570), ExactPosition(571)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(675), ExactPosition(728)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(675), ExactPosition(728)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(740), ExactPosition(1070)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(1006), ExactPosition(1007)), type='Bond'), SeqFeature(FeatureLocation(ExactPosition(1110), ExactPosition(1111)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1152), ExactPosition(1210)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(1162), ExactPosition(1215)), type='Region'), SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1162), ExactPosition(1163)), FeatureLocation(ExactPosition(1172), ExactPosition(1177)), FeatureLocation(ExactPosition(1180), ExactPosition(1181)), FeatureLocation(ExactPosition(1209), ExactPosition(1210))], 'order'), type='Site', location_operator='order'), SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1163), ExactPosition(1164)), FeatureLocation(ExactPosition(1165), ExactPosition(1166)), FeatureLocation(ExactPosition(1176), ExactPosition(1177)), FeatureLocation(ExactPosition(1179), ExactPosition(1180)), FeatureLocation(ExactPosition(1184), ExactPosition(1185)), FeatureLocation(ExactPosition(1187), ExactPosition(1188)), FeatureLocation(ExactPosition(1211), ExactPosition(1212)), FeatureLocation(ExactPosition(1214), ExactPosition(1215))], 'order'), type='Site', location_operator='order'), SeqFeature(FeatureLocation(ExactPosition(1271), ExactPosition(1272)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(1329), ExactPosition(1330)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1330), ExactPosition(1331)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1342), ExactPosition(1343)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1344), ExactPosition(1345)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1437), ExactPosition(1438)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1483), ExactPosition(1579)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(1487), ExactPosition(1488)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1594), ExactPosition(1595)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1597), ExactPosition(1598)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1598), ExactPosition(1653)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(1602), ExactPosition(1603)), type='Site'), SeqFeature(FeatureLocation(ExactPosition(1607), ExactPosition(1659)), type='Region'), SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1609), ExactPosition(1610)), FeatureLocation(ExactPosition(1614), ExactPosition(1615)), FeatureLocation(ExactPosition(1627), ExactPosition(1628)), FeatureLocation(ExactPosition(1631), ExactPosition(1632)), FeatureLocation(ExactPosition(1636), ExactPosition(1637)), FeatureLocation(ExactPosition(1639), ExactPosition(1640)), FeatureLocation(ExactPosition(1654), ExactPosition(1655)), FeatureLocation(ExactPosition(1657), ExactPosition(1658))], 'order'), type='Site', location_operator='order'), SeqFeature(FeatureLocation(ExactPosition(1621), ExactPosition(1690)), type='Region'), SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1623), ExactPosition(1629)), FeatureLocation(ExactPosition(1632), ExactPosition(1633)), FeatureLocation(ExactPosition(1634), ExactPosition(1635)), FeatureLocation(ExactPosition(1637), ExactPosition(1638))], 'order'), type='Site', location_operator='order'), SeqFeature(FeatureLocation(ExactPosition(1665), ExactPosition(1666)), type='Site')]\n"
     ]
    }
   ],
   "source": [
    "print(record.features)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Note that a more typical use would be to save the sequence data to a local file, and then parse it with Bio.SeqIO. This can save you having to re-download the same file repeatedly while working on your script, and places less load on the NCBI’s servers. For example:\n",
    "\n",
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "Entrez.email = \"A.N.Other@example.com\"  # Always tell NCBI who you are\n",
    "filename = \"EU490707.gbk\"\n",
    "if not os.path.isfile(filename):\n",
    "    # Downloading...\n",
    "    net_handle = Entrez.efetch(db=\"nucleotide\", id=\"EU490707\", rettype=\"gb\", retmode=\"text\")\n",
    "    out_handle = open(filename, \"w\")\n",
    "    out_handle.write(net_handle.read())\n",
    "    out_handle.close()\n",
    "    net_handle.close()\n",
    "    print(\"Saved\")\n",
    "\n",
    "print(\"Parsing...\")\n",
    "record = SeqIO.read(filename, \"genbank\")\n",
    "print(record)\n",
    "#To get the output in XML format, which you can parse using the Bio.Entrez.read() function, use retmode=\"xml\":\n",
    "\n",
    ">>> from Bio import Entrez\n",
    ">>> Entrez.email = \"A.N.Other@example.com\"  # Always tell NCBI who you are\n",
    ">>> handle = Entrez.efetch(db=\"nucleotide\", id=\"EU490707\", retmode=\"xml\")\n",
    ">>> record = Entrez.read(handle)\n",
    ">>> handle.close()\n",
    ">>> record[0][\"GBSeq_definition\"]\n",
    "'Selenipedium aequinoctiale maturase K (matK) gene, partial cds; chloroplast'\n",
    ">>> record[0][\"GBSeq_source\"]\n",
    "'chloroplast Selenipedium aequinoctiale'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "5.1.2  Iterating over the records in a sequence file\n",
    "In the above examples, we have usually used a for loop to iterate over all the records one by one. You can use the for loop with all sorts of Python objects (including lists, tuples and strings) which support the iteration interface.\n",
    "\n",
    "The object returned by Bio.SeqIO is actually an iterator which returns SeqRecord objects. You get to see each record in turn, but once and only once. The plus point is that an iterator can save you memory when dealing with large files.\n",
    "\n",
    "Instead of using a for loop, can also use the next() function on an iterator to step through the entries, like this:\n",
    "\n",
    "from Bio import SeqIO\n",
    "record_iterator = SeqIO.parse(\"ls_orchid.fasta\", \"fasta\")\n",
    "\n",
    "first_record = next(record_iterator)\n",
    "print(first_record.id)\n",
    "print(first_record.description)\n",
    "\n",
    "second_record = next(record_iterator)\n",
    "print(second_record.id)\n",
    "print(second_record.description)\n",
    "Note that if you try to use next() and there are no more results, you’ll get the special StopIteration exception.\n",
    "\n",
    "One special case to consider is when your sequence files have multiple records, but you only want the first one. In this situation the following code is very concise:\n",
    "\n",
    "from Bio import SeqIO\n",
    "first_record = next(SeqIO.parse(\"ls_orchid.gbk\", \"genbank\"))  \n",
    "A word of warning here – using the next() function like this will silently ignore any additional records in the file. If your files have one and only one record, like some of the online examples later in this chapter, or a GenBank file for a single chromosome, then use the new Bio.SeqIO.read() function instead. This will check there are no extra unexpected records present.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Extracting data\n",
    "#The SeqRecord object and its annotation structures are described more fully in Chapter 4. As an example of how annotations are stored, we’ll look at the output from parsing the first record in the GenBank file ls_orchid.gbk.\n",
    "http://biopython.org/DIST/docs/tutorial/Tutorial.html#chapter:Bio.SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'NoneType' object is not callable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-29-64d09b776aca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mBio\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mrecord_iterator\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"genbank\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mfirst_record\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrecord_iterator\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[1;32mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfirst_record\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\David Brown\\Anaconda2\\lib\\site-packages\\Bio\\SeqIO\\__init__.pyc\u001b[0m in \u001b[0;36mread\u001b[0;34m(handle, format, alphabet)\u001b[0m\n\u001b[1;32m    700\u001b[0m     \u001b[0miterator\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mparse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mformat\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0malphabet\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    701\u001b[0m     \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0mfirst\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0miterator\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m     \u001b[1;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m         \u001b[0mfirst\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\David Brown\\Anaconda2\\lib\\site-packages\\Bio\\SeqIO\\__init__.pyc\u001b[0m in \u001b[0;36mparse\u001b[0;34m(handle, format, alphabet)\u001b[0m\n\u001b[1;32m    635\u001b[0m             \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"Unknown format '%s'\"\u001b[0m \u001b[1;33m%\u001b[0m \u001b[0mformat\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    636\u001b[0m         \u001b[1;31m# This imposes some overhead... wait until we drop Python 2.4 to fix it\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 637\u001b[0;31m         \u001b[1;32mfor\u001b[0m \u001b[0mr\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mi\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    638\u001b[0m             \u001b[1;32myield\u001b[0m \u001b[0mr\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    639\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\David Brown\\Anaconda2\\lib\\site-packages\\Bio\\GenBank\\Scanner.pyc\u001b[0m in \u001b[0;36mparse_records\u001b[0;34m(self, handle, do_features)\u001b[0m\n\u001b[1;32m    487\u001b[0m         \u001b[1;31m# This is a generator function\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    488\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[0mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 489\u001b[0;31m             \u001b[0mrecord\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    490\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mrecord\u001b[0m \u001b[1;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    491\u001b[0m                 \u001b[1;32mbreak\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\David Brown\\Anaconda2\\lib\\site-packages\\Bio\\GenBank\\Scanner.pyc\u001b[0m in \u001b[0;36mparse\u001b[0;34m(self, handle, do_features)\u001b[0m\n\u001b[1;32m    471\u001b[0m                                     feature_cleaner=FeatureValueCleaner())\n\u001b[1;32m    472\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 473\u001b[0;31m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    474\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    475\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\David Brown\\Anaconda2\\lib\\site-packages\\Bio\\GenBank\\Scanner.pyc\u001b[0m in \u001b[0;36mfeed\u001b[0;34m(self, handle, consumer, do_features)\u001b[0m\n\u001b[1;32m    428\u001b[0m         \u001b[1;31m# equivalent Bio.GenBank._FeatureConsumer methods are called...\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    429\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_handle\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 430\u001b[0;31m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind_start\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    431\u001b[0m             \u001b[1;31m# Could not find (another) record\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    432\u001b[0m             \u001b[0mconsumer\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdata\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\David Brown\\Anaconda2\\lib\\site-packages\\Bio\\GenBank\\Scanner.pyc\u001b[0m in \u001b[0;36mfind_start\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     84\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m     85\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m---> 86\u001b[0;31m                 \u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     87\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m     88\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdebug\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'NoneType' object is not callable"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "record_iterator = SeqIO.parse(\"ls_orchid.gbk\", \"genbank\")\n",
    "first_record = next(record_iterator)\n",
    "print(first_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "That should give something like this:\n",
    "\n",
    "ID: Z78533.1\n",
    "Name: Z78533\n",
    "Description: C.irapeanum 5.8S rRNA gene and ITS1 and ITS2 DNA.\n",
    "Number of features: 5\n",
    "/sequence_version=1\n",
    "/source=Cypripedium irapeanum\n",
    "/taxonomy=['Eukaryota', 'Viridiplantae', 'Streptophyta', ..., 'Cypripedium']\n",
    "/keywords=['5.8S ribosomal RNA', '5.8S rRNA gene', ..., 'ITS1', 'ITS2']\n",
    "/references=[...]\n",
    "/accessions=['Z78533']\n",
    "/data_file_division=PLN\n",
    "/date=30-NOV-2006\n",
    "/organism=Cypripedium irapeanum\n",
    "/gi=2765658\n",
    "Seq('CGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTGATGAGACCGTGG...CGC', IUPACAmbiguousDNA())\n",
    "This gives a human readable summary of most of the annotation data for the SeqRecord. For this example we’re going to use the .annotations attribute which is just a Python dictionary. The contents of this annotations dictionary were shown when we printed the record above. You can also print them out directly:\n",
    "\n",
    "print(first_record.annotations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "record_iterator = SeqIO.parse(\"ls_orchid.gbk\", \"genbank\")\n",
    "first_record = next(record_iterator)\n",
    "print(first_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "55\n",
      "[SeqFeature(FeatureLocation(ExactPosition(13), ExactPosition(14)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(16), ExactPosition(17)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(17), ExactPosition(59)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(18), ExactPosition(60)), type='Region'), SeqFeature(FeatureLocation(ExactPosition(23), ExactPosition(24)), type='Region')]\n"
     ]
    }
   ],
   "source": [
    "#I can do this for my SeqRecords from Entrez\n",
    "#print(record.features)\n",
    "print(len(record.features))\n",
    "print(record.features[5:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type: Region\n",
      "location: [292:343]\n",
      "qualifiers:\n",
      "    Key: experiment, Value: ['experimental evidence, no additional details recorded']\n",
      "    Key: note, Value: ['PHD-type 1. {ECO:0000255|PROSITE-ProRule:PRU00146}.']\n",
      "    Key: region_name, Value: ['Zinc finger region']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Print a specific feature within the record\n",
    "f=record.features[14] #In this case PHD1\n",
    "print(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('LYVCMFCGRGNNEDKLLLCDGCDDSYHTFCLLPPLPDVPKGDWRCPKCVAE', IUPACProtein())"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#extract the sequence of a specific feature within the record\n",
    "f.extract(record.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MASVGPGGYAAEFVPPPECPVFEPSWEEFTDPLSFIGRIRPFAEKTGICKIRPPKDWQPPFACEVKTFRFTPRVQRLNELEAMTRVRLDFLDQLAKFWELQGSTLKIPVVERKILDLYALSKIVASKGGFEIVTKEKKWSKVGSRLGYLPGKGTGSLLKSHYERILYPYELFQSGVSLMGVQMPDLDLKEKVEAEVLSTDIQPSPERGTRMNIPPKRTRRVKSQSDSGEVNRNTELKKLQIFGAGPKVVGLAVGAKDKEDEVTRRRKVTNRSDAFNMQMRQRKGTLSVNFVDLYVCMFCGRGNNEDKLLLCDGCDDSYHTFCLLPPLPDVPKGDWRCPKCVAEECNKPREAFGFEQAVREYTLQSFGEMADNFKSDYFNMPVHMVPTELVEKEFWRLVSSIEEDVIVEYGADISSKDFGSGFPKKDGQRKMLPEEEEYALSGWNLNNMPVLEQSVLAHINVDISGMKVPWLYVGMCFSSFCWHIEDHWSYSINYLHWGEPKTWYGVPSHAAEQLEEVMRELAPELFESQPDLLHQLVTIMNPNVLMEHGVPVYRTNQCAGEFVVTFPRAYHSGFNQGYNFAEAVNFCTADWLPIGRQCVNHYRRLRRHCVFSHEELIFKMAADPECLDVGLAAMVCKELTLMTEEETRLRESVVQMGVVMSEEEVFELVPDDERQCSACRTTCFLSALTCSCNPERLVCLYHPTDLCSCPMQNKCLRYRYPLEDLPSLLYGVKVRAQSYDTWVNRVTEALSASFNHKKDLIELRVMLEDAEDRKYPENDLFRKLRDAVKEAETCGSVAQLLLSKKQKHRQSSDSGKTRTKLTVEELKAFVQQLVSLPCVISQTRQVKNLLDDVEEFHERAQEAMMDETPDSSKLQMLIDMGSSLYVELPELPRLKQELQQARWLDEVRLTLSDPQQVTLDVMKKLIDSGVGLAPHHAVEKAMAELQELLTVSERWEEKAKVCLQARPRHSMANLENIVNEAKNIPAFLPNVLSLKEALQKAREWTAKVEAIQSGNNYAYLEQLESLSAKGRPIPVRLDALPQVESQVAAARAWRERTGRTFLKKNSSHTLLQVLSPRTDIGVYGSGKNRRKKVKEIIEKEKEKDLDLEPLSDLEEGLEESRDTAMVVAVFKEREQKEIEAMHSLRAANLAKMTIVERIEEVKFCICRKTASGFMLQCELCKDWFHNSCVPLPKSSSQKKGSSWQAKDVKFLCPLCMRSRRPRLETILSLLVSLQKLPVRLPEGEALQCLTERAMSWQDKARQALATDELSSALAKLSVLSQRMVEQAAREKTEKIISAELQKAAANPDLQGHLPSFQQSAFNRVVSSVSSSPHQTMDYDDEETDSDEDIRETYGYDMKDTASVKSSSSLEPNLFCDEEIPIKSEEVVTHMWTAPSFCAEHAYSSASKSCSQGSSTPRKQPRKSPLVPRSLEPPVLELSPGAKAQLEELMMVGDLLEVSLDETQHIWRILQATHPPSEDRFLHIMEDDSIEEKPLKMKGKDSSEKKRKRKLEKVEQLFGEGKQKTKELKKIDKPKKKKLKLNVDKSKELNKLAKKLAKEEERKKKKEKAAAAKVELVKESTEKKRERKVLDIPSKYDWSGAEESDDENAVCAAQNCQRPCKDKVDWVQCDGGCDEWFHQVCVGVSAEMAENEDYICINCAKKQGPDSPGQAPPPPFLMSYKLPMEDLKETS\n",
      "A\n",
      "V\n",
      "P\n",
      "ECPVFEPSWEEFTDPLSFIGRIRPFAEKTGICKIRPPKDWQP\n",
      "CPVFEPSWEEFTDPLSFIGRIRPFAEKTGICKIRPPKDWQPP\n",
      "P\n",
      "TRVRLDFLDQLAKFWELQGSTLKIPVVERKILDLYALSKIVASKGGFEIVTKEKKWSKVGSRLGYLPGKGTGSLLKSHYERILYPYELFQS\n",
      "RVRLDFLDQLAKFWELQGSTLKIPVVERKILDLYALSKIVASKGGFEIVTKEKKWSKVGSRLGYLPGKGTGSLLKSHYERILYPYE\n",
      "LYVCMFCGRGNNEDKLLLCDGCDDSYHTFCLLPPLPDVPKGDWRCPKCVAE\n",
      "VCMFCGRGNNEDKLLLCDGCDDSYHTFCLLPPLPDVPKGDWRCPKCVAE\n",
      "GSGFP\n",
      "EYALSGWNLNNMPVLEQSVLAHINVDISGMKVPWLYVGMCFSSFCWHIEDHWSYSINYLHWGEPKTWYGVPSHAAEQLEEVMRELAPELFESQPDLLHQLVTIMNPNVLMEHGVPVYRTNQCAGEFVVTFPRAYHSGFNQGYNFAEAVNFCTADWLPIGRQCVNHYR\n",
      "WLYVGMCFSSFCWHIEDHWSYSINYLHWGEPKTWYGVPSHAAEQLEEVMRELAPELFESQPDLLHQLVTIMNPNVLMEHGVPVYRTNQCAGEFVVTFPRAYHSGFNQGYNFAEAVNF\n",
      "V\n",
      "CSACRTTCFLSALTCSCNPERLVCLYHPTDLCSCPMQNKCLRYRYPLEDLPSL\n",
      "CSACRTTCFLSALTCSCNPERLVCLYHPTDLCSCPMQNKCLRYRYPLEDLPSL\n",
      "TWVNRVTEALSASFNHKKDLIELRVMLEDAEDRKYPENDLFRKLRDAVKEAETCGSVAQLLLSKKQKHRQSSDSGKTRTKLTVEELKAFVQQLVSLPCVISQTRQVKNLLDDVEEFHERAQEAMMDETPDSSKLQMLIDMGSSLYVELPELPRLKQELQQARWLDEVRLTLSDPQQVTLDVMKKLIDSGVGLAPHHAVEKAMAELQELLTVSERWEEKAKVCLQARPRHSMANLENIVNEAKNIPAFLPNVLSLKEALQKAREWTAKVEAIQSGNNYAYLEQLESLSAKGRPIPVRLDALPQVESQVAAARAWRERTGRTFLKKNSSHTL\n",
      "TIVERIEEVKFCICRKTASGFMLQCELCKDWFHNSCVPLPKSSSQKKGSSWQAKDVKF\n",
      "FCICRKTASGFMLQCELCKDWFHNSCVPLPKSSSQKKGSSWQAKDVKFLCPLC\n",
      "A\n",
      "MEDDSIEEKPLKMKGKDSSEKKRKRKLEKVEQLFGEGKQKTKELKKIDKPKKKKLKLNVDKSKELNKLAKKLAKEEERKKKKEKAAAAKVELVKES\n",
      "GAEESDDENAVCAAQNCQRPCKDKVDWVQCDGGCDEWFHQVCVGVSAEMAENEDY\n",
      "AVCAAQNCQRPCKDKVDWVQCDGGCDEWFHQVCVGVSAEMAENEDYICINCA\n",
      "KVDWVQCDGGCDEWFHQVCVGVSAEMAENEDYICINCAKKQGPDSPGQAPPPPFLMSYKLPMEDLKETS\n"
     ]
    }
   ],
   "source": [
    "#Can I extract the sequences of all domains/regions?\n",
    "s = record.features\n",
    "for f in s:\n",
    "    if f.type == 'Region':\n",
    "        #n=f.region_name #How do I get the names of the regions?\n",
    "        r=f.extract(record.seq)\n",
    "        #print(n)\n",
    "        print(r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#To access data\n",
    "Bio.SeqIO.index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Can I extract the sequences of all domains/regions?\n",
    "s = record.features\n",
    "for f in s:\n",
    "    if f.type == 'Region':\n",
    "        #re.finditer(\"PHD\", f.qualifiers[\"note\"], flags=0)\n",
    "        i=(f.qualifiers)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'region_name'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-65-59a0d50f5bb2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mre\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfinditer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"ARID\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mqualifiers\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"region_name\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m: 'region_name'"
     ]
    }
   ],
   "source": [
    "re.finditer(\"ARID\", f.qualifiers[\"region_name\"], flags=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "i=re.search(\"c\", \"abcdef\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'OrderedDict' object has no attribute 'note'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-81-dbfd73cbd5fd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mi\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mnote\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m: 'OrderedDict' object has no attribute 'note'"
     ]
    }
   ],
   "source": [
    "i.note"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
